{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Question 2 Workflow\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Workfolow Summary"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/data_tools/df_processing.py:4: TqdmExperimentalWarning: Using `tqdm.autonotebook.tqdm` in notebook mode. Use `tqdm.tqdm` instead to force console mode (e.g. in jupyter console)\n",
      "  from tqdm.autonotebook import tqdm\n"
     ]
    }
   ],
   "source": [
    "# Import pandas and biothings explorers modules\n",
    "import pandas as pandas\n",
    "from biothings_explorer.query.predict import Predict\n",
    "from biothings_explorer.query.visualize import display_graph\n",
    "from biothings_explorer.user_query_dispatcher import FindConnection\n",
    "from biothings_explorer.hint import Hint\n",
    "import nest_asyncio\n",
    "nest_asyncio.apply()\n",
    "%matplotlib inline\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\") "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Set Parameters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "max_direct = 20\n",
    "max_indirect = 50"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'MONDO': 'MONDO:0100096',\n",
       " 'DOID': 'DOID:0080600',\n",
       " 'name': 'COVID-19',\n",
       " 'primary': {'identifier': 'MONDO',\n",
       "  'cls': 'Disease',\n",
       "  'value': 'MONDO:0100096'},\n",
       " 'display': 'MONDO(MONDO:0100096) DOID(DOID:0080600) name(COVID-19)',\n",
       " 'type': 'Disease'}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "disease_name = 'COVID-19'\n",
    "ht = Hint()\n",
    "disease = ht.query(disease_name)['Disease'][0]\n",
    "disease"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get Symptoms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests \n",
    "\n",
    "symptoms = ([ 'fever','cough', 'hemoptysis', \n",
    "'shortness of breath', 'myalgia', 'fatigue', 'sore throat', \n",
    "'nausea', 'vomiting', 'diarrhea', 'conjunctivitis', 'anorexia',\n",
    "'headache', 'leukopenia', 'eosinopenia', 'neutrophilia', 'elevated liver enzymes', 'C-reactive protein', 'ferritin',\n",
    "'hyaline membrane', 'alveolar lesions', 'prominent hyaline membranes', 'alveolar damage',\n",
    " 'acute myocardial injury', ' myocarditis', 'arrhythmias','cardiac dysfunction', 'encephalopathy'])\n",
    "\n",
    "\n",
    "symptom_dict = {}\n",
    "\n",
    "for symptom_name in symptoms: \n",
    "    try: \n",
    "        symptom = ht.query(symptom_name)['PhenotypicFeature'][0]\n",
    "#         print(symptom['name'])\n",
    "        pd = Predict(\n",
    "            input_objs=[symptom],\n",
    "            intermediate_nodes =[None], \n",
    "            output_types = ['Gene', 'SequenceVariant', 'ChemicalSubstance', 'Disease', \n",
    "                'MolecularActivity', 'BiologicalProcess', 'CellularComponent', \n",
    "                'Pathway', 'AnatomicalEntity', 'PhenotypicFeature'], \n",
    "            config={\"expand\": False}\n",
    "        )\n",
    "        pd.connect(verbose=False)\n",
    "        df = pd.display_table_view()\n",
    "        edges_out = df.shape[0]\n",
    "        symptom_dict[symptom['HP']] = {\n",
    "            'names': [symptom['name'].upper()],\n",
    "            'frequency': 'Unknown', \n",
    "            'edges_out' : edges_out\n",
    "        }\n",
    "        r = requests.get('https://biothings.ncats.io/hpo/phenotype/' + str(symptom['HP']))\n",
    "        res = r.json()\n",
    "#         print(res)\n",
    "        if('synonym' in res):\n",
    "            if('exact' in res['synonym']):\n",
    "                for name in res['synonym']['exact']:\n",
    "                    if name.upper() not in symptom_dict[res['_id']]['names']: \n",
    "#                         print(name.upper())\n",
    "                        symptom_dict[res['_id']]['names'].append(name.upper())\n",
    "    except:\n",
    "        print(symptom_name + \" failed\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>names</th>\n",
       "      <th>frequency</th>\n",
       "      <th>edges_out</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HP:0011134</th>\n",
       "      <td>[LOW-GRADE FEVER, MILD FEVER]</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>9553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP:0002105</th>\n",
       "      <td>[HEMOPTYSIS, COUGHING UP BLOOD, HAEMOPTYSIS]</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>46624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP:0003326</th>\n",
       "      <td>[MYALGIA, MUSCLE ACHE, MUSCLE PAIN]</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>116332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP:0500039</th>\n",
       "      <td>[CONJUNCTIVAL CICATRIZATION, CICATRICIAL CONJU...</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP:0002039</th>\n",
       "      <td>[ANOREXIA]</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>125701</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP:0011897</th>\n",
       "      <td>[NEUTROPHILIA, INCREASED BLOOD NEUTROPHIL COUNTS]</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>6354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP:0032436</th>\n",
       "      <td>[ABNORMAL C-REACTIVE PROTEIN LEVEL]</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP:0040133</th>\n",
       "      <td>[ABNORMAL SERUM FERRITIN, ABNORMAL PLASMA FERR...</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP:0005115</th>\n",
       "      <td>[SUPRAVENTRICULAR ARRHYTHMIA, ARRHYTHMIAS, SUP...</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>11784</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                        names frequency  \\\n",
       "HP:0011134                      [LOW-GRADE FEVER, MILD FEVER]   Unknown   \n",
       "HP:0002105       [HEMOPTYSIS, COUGHING UP BLOOD, HAEMOPTYSIS]   Unknown   \n",
       "HP:0003326                [MYALGIA, MUSCLE ACHE, MUSCLE PAIN]   Unknown   \n",
       "HP:0500039  [CONJUNCTIVAL CICATRIZATION, CICATRICIAL CONJU...   Unknown   \n",
       "HP:0002039                                         [ANOREXIA]   Unknown   \n",
       "HP:0011897  [NEUTROPHILIA, INCREASED BLOOD NEUTROPHIL COUNTS]   Unknown   \n",
       "HP:0032436                [ABNORMAL C-REACTIVE PROTEIN LEVEL]   Unknown   \n",
       "HP:0040133  [ABNORMAL SERUM FERRITIN, ABNORMAL PLASMA FERR...   Unknown   \n",
       "HP:0005115  [SUPRAVENTRICULAR ARRHYTHMIA, ARRHYTHMIAS, SUP...   Unknown   \n",
       "\n",
       "            edges_out  \n",
       "HP:0011134       9553  \n",
       "HP:0002105      46624  \n",
       "HP:0003326     116332  \n",
       "HP:0500039          3  \n",
       "HP:0002039     125701  \n",
       "HP:0011897       6354  \n",
       "HP:0032436        257  \n",
       "HP:0040133        440  \n",
       "HP:0005115      11784  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# symptom_dict\n",
    "disease_symptom_df = pandas.DataFrame.from_dict(symptom_dict, orient='index')\n",
    "disease_symptom_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "## also save disease symptoms as a list\n",
    "disease_symptoms = []\n",
    "for x in disease_symptom_df['names']: \n",
    "    disease_symptoms = disease_symptoms + x\n",
    "# disease_symptoms"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Disease -> Gene -> [Phenotypic Feature, Disease, Biological Process]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 165, in callSingleAPI\n",
      "    async with session.get(\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/aiohttp/client.py\", line 1012, in __aenter__\n",
      "    self._resp = await self._coro\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/aiohttp/client.py\", line 504, in _request\n",
      "    await resp.start(conn)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/aiohttp/client_reqrep.py\", line 847, in start\n",
      "    message, payload = await self._protocol.read()  # type: ignore  # noqa\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/aiohttp/streams.py\", line 591, in read\n",
      "    await self._waiter\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/futures.py\", line 260, in __await__\n",
      "    yield self  # This tells Task to wait for completion.\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/tasks.py\", line 349, in __wakeup\n",
      "    future.result()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/futures.py\", line 178, in result\n",
      "    raise self._exception\n",
      "aiohttp.client_exceptions.ClientOSError: [Errno 1] [SSL: APPLICATION_DATA_AFTER_CLOSE_NOTIFY] application data after close notify (_ssl.c:2745)\n",
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 165, in callSingleAPI\n",
      "    async with session.get(\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/aiohttp/client.py\", line 1012, in __aenter__\n",
      "    self._resp = await self._coro\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/aiohttp/client.py\", line 504, in _request\n",
      "    await resp.start(conn)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/aiohttp/client_reqrep.py\", line 847, in start\n",
      "    message, payload = await self._protocol.read()  # type: ignore  # noqa\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/aiohttp/streams.py\", line 591, in read\n",
      "    await self._waiter\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/futures.py\", line 260, in __await__\n",
      "    yield self  # This tells Task to wait for completion.\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/tasks.py\", line 349, in __wakeup\n",
      "    future.result()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/futures.py\", line 178, in result\n",
      "    raise self._exception\n",
      "aiohttp.client_exceptions.ClientOSError: [Errno 1] [SSL: APPLICATION_DATA_AFTER_CLOSE_NOTIFY] application data after close notify (_ssl.c:2745)\n",
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 165, in callSingleAPI\n",
      "    async with session.get(\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/aiohttp/client.py\", line 1012, in __aenter__\n",
      "    self._resp = await self._coro\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/aiohttp/client.py\", line 504, in _request\n",
      "    await resp.start(conn)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/aiohttp/client_reqrep.py\", line 847, in start\n",
      "    message, payload = await self._protocol.read()  # type: ignore  # noqa\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/aiohttp/streams.py\", line 591, in read\n",
      "    await self._waiter\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/futures.py\", line 260, in __await__\n",
      "    yield self  # This tells Task to wait for completion.\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/tasks.py\", line 349, in __wakeup\n",
      "    future.result()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/futures.py\", line 178, in result\n",
      "    raise self._exception\n",
      "aiohttp.client_exceptions.ClientOSError: [Errno 1] [SSL: APPLICATION_DATA_AFTER_CLOSE_NOTIFY] application data after close notify (_ssl.c:2745)\n",
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 165, in callSingleAPI\n",
      "    async with session.get(\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/aiohttp/client.py\", line 1012, in __aenter__\n",
      "    self._resp = await self._coro\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/aiohttp/client.py\", line 504, in _request\n",
      "    await resp.start(conn)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/aiohttp/client_reqrep.py\", line 847, in start\n",
      "    message, payload = await self._protocol.read()  # type: ignore  # noqa\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/aiohttp/streams.py\", line 591, in read\n",
      "    await self._waiter\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/futures.py\", line 260, in __await__\n",
      "    yield self  # This tells Task to wait for completion.\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/tasks.py\", line 349, in __wakeup\n",
      "    future.result()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/futures.py\", line 178, in result\n",
      "    raise self._exception\n",
      "aiohttp.client_exceptions.ClientOSError: [Errno 1] [SSL: APPLICATION_DATA_AFTER_CLOSE_NOTIFY] application data after close notify (_ssl.c:2745)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API call to BioLink API with input 128989 failed with unknown response\n",
      "API call to BioLink API with input R-MMU-8949664 failed with unknown response\n",
      "API call to BioLink API with input HP:0002448 failed with unknown response\n",
      "API call to BioLink API with input HP:0002480 failed with unknown response\n",
      "API call to Automat CORD19 Scigraph API with input 1401 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Automat CORD19 Scigraph API with input 59272 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Automat PHAROS API with input 6651 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Automat CORD19 Scibite API with input 59272 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Automat CORD19 Scibite API with input 7113 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Automat PHAROS API with input 1401 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Automat CORD19 Scibite API with input 6651 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Automat CORD19 Scigraph API with input 6651 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Automat CORD19 Scibite API with input 1401 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Automat CORD19 Scigraph API with input 7113 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Automat PHAROS API with input 7113 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Automat PHAROS API with input 59272 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Molecular Data Provider API with input 11183 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Molecular Data Provider API with input 2367 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Molecular Data Provider API with input 13557 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Genetics Provider API with input 1401 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Genetics Provider API with input 59272 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Genetics Provider API with input 7113 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Genetics Provider API with input 6651 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to CORD Gene API with input ['11892', '9281', '19679', '79', '2707', '2367', '10618', '399', '27954', '6915', '5438', '1116', '6018', '5414', '9208', '6898', '13523', '8568', '11183', '10637', '16884', '30615', '3541', '6741', '13557', '1706', '6008', '4552', '11782', '4989', '5981', '11876', '11947', '17697', '10627', '6023', '338', '5992', '1678', '6001', '27960', '336', '5417', '6025', '16885', '9958', '4113', '5962', '195', '3535', '2537', '5434', '2524', '3009'] failed with timeout error. Current timeout limit is 20 seconds\n",
      "API call to CORD Gene API with input ['11892', '9281', '19679', '79', '2707', '2367', '10618', '399', '27954', '6915', '5438', '1116', '6018', '5414', '9208', '6898', '13523', '8568', '11183', '10637', '16884', '30615', '3541', '6741', '13557', '1706', '6008', '4552', '11782', '4989', '5981', '11876', '11947', '17697', '10627', '6023', '338', '5992', '1678', '6001', '27960', '336', '5417', '6025', '16885', '9958', '4113', '5962', '195', '3535', '2537', '5434', '2524', '3009'] failed with timeout error. Current timeout limit is 20 seconds\n"
     ]
    }
   ],
   "source": [
    "pd = Predict(\n",
    "    input_objs=[disease],\n",
    "    intermediate_nodes =['Gene'], \n",
    "    output_types =['PhenotypicFeature', 'Disease', 'BiologicalProcess'], \n",
    "    config={\"expand\": False}\n",
    ")\n",
    "pd.connect(verbose=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(34593, 19)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.display_table_view()\n",
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(9, 19)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# filter gene -> symptoms table to only include rows where symptoms match disease symptoms\n",
    "indices_with_symptom_outputs = [i for i, val in enumerate([x.upper() for x in list(df['output_label'])]) if val in disease_symptoms]\n",
    "direct_relevant_genes_to_symptoms_df = df.iloc[indices_with_symptom_outputs]\n",
    "direct_relevant_genes_to_symptoms_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'TNF': 4, 'IL2': 1, 'IL1B': 1, 'IFNA1': 1, 'TNNI3': 1, 'IL6': 1}"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "direct_genes = \tlist(direct_relevant_genes_to_symptoms_df[\"node1_label\"])\n",
    "direct_gene_dict = {x:direct_genes.count(x) for x in direct_genes}\n",
    "direct_gene_dict = dict(sorted(direct_gene_dict.items(), key=lambda x: x[1], reverse=True)[:max_direct])\n",
    "direct_gene_dict"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Disease X -> Intermediate Node -> Gene ->  [Phenotypic Feature, Disease, Biological Process]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "query_config = {\n",
    "    \"filters\": [\n",
    "        {\n",
    "        },\n",
    "        {\n",
    "            \"nodeDegree\": {\n",
    "                \">\" : 100\n",
    "            }\n",
    "        }\n",
    "    ],\n",
    "    \"annotate\": [\"ngd\",\"nodeDegree\"],\n",
    "    \"expand\": False\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==========\n",
      "========== QUERY PARAMETER SUMMARY ==========\n",
      "==========\n",
      "\n",
      "Your query have 1 input nodes, including COVID-19 .... And BTE will find paths that connect your input nodes to your output types ['PhenotypicFeature', 'Disease', 'BiologicalProcess']. Paths will contain 2 intermediate nodes.\n",
      "\n",
      "Intermediate node #1 will have these type constraints: ['Gene', 'SequenceVariant', 'ChemicalSubstance', 'Disease', 'MolecularActivity', 'BiologicalProcess', 'CellularComponent', 'Pathway', 'AnatomicalEntity', 'PhenotypicFeature']\n",
      "Intermediate node #2 will have these type constraints: Gene\n",
      "\n",
      "========== Query # 1 ==========\n",
      "\n",
      "==== Step #1: Query Path Planning ====\n",
      "\n",
      "Input Types: Disease\n",
      "Output Types: ['Gene', 'SequenceVariant', 'ChemicalSubstance', 'Disease', 'MolecularActivity', 'BiologicalProcess', 'CellularComponent', 'Pathway', 'AnatomicalEntity', 'PhenotypicFeature']\n",
      "Predicates: None\n",
      "\n",
      "BTE found 12 APIs based on SmartAPI Meta-KG.\n",
      "\n",
      "API 1. Ontology Lookup Service API (1 API calls)\n",
      "API 2. Automat CORD19 Scibite API (3 API calls)\n",
      "API 3. mydisease.info API (2 API calls)\n",
      "API 4. Automat CORD19 Scigraph API (3 API calls)\n",
      "API 5. MGIgene2phenotype API (1 API calls)\n",
      "API 6. TCGA Mutation Frequency KP API (1 API calls)\n",
      "API 7. Automat PHAROS API (3 API calls)\n",
      "API 8. DISEASES API (1 API calls)\n",
      "API 9. BioLink API (4 API calls)\n",
      "API 10. Molecular Data Provider API (1 API calls)\n",
      "API 11. Automat HMDB API (2 API calls)\n",
      "API 12. CORD Disease API (8 API calls)\n",
      "\n",
      "\n",
      "==== Step #2: Query path execution ====\n",
      "\n",
      "NOTE: API requests are dispatched in parallel, so the list of APIs below is ordered by query time.\n",
      "\n",
      "API 3.1: http://mydisease.info/v1/query?fields=mondo.parents (POST -d q=MONDO:0100096&scopes=mondo.mondo)\n",
      "API 3.1 mydisease.info API: 1 hits\n",
      "API 3.2: http://mydisease.info/v1/query?fields=mondo.descendants (POST -d q=MONDO:0100096&scopes=mondo.mondo)\n",
      "API 3.2 mydisease.info API: 0 hits\n",
      "API 12.1: https://biothings.ncats.io/cord_disease/query?fields=associated_with (POST -d q=DOID:0080600&scopes=doid)\n",
      "API 12.1 CORD Disease API: 0 hits\n",
      "API 12.2: https://biothings.ncats.io/cord_disease/query?fields=associated_with (POST -d q=DOID:0080600&scopes=doid)\n",
      "API 12.2 CORD Disease API: 0 hits\n",
      "API 12.3: https://biothings.ncats.io/cord_disease/query?fields=associated_with (POST -d q=DOID:0080600&scopes=doid)\n",
      "API 12.3 CORD Disease API: 0 hits\n",
      "API 6.1: https://biothings.ncats.io/tcga_mut_freq_kp/query?fields=association.freq_by_sample,subject.SYMBOL&q=object.id:\"MONDO:0100096\"%20AND%20association.freq_by_sample:>0.1&size=1000&sort=-association.freq_by_sample\n",
      "API 6.1 TCGA Mutation Frequency KP API: 0 hits\n",
      "API 12.4: https://biothings.ncats.io/cord_disease/query?fields=associated_with (POST -d q=DOID:0080600&scopes=doid)\n",
      "API 12.4 CORD Disease API: 0 hits\n",
      "API 9.1: https://api.monarchinitiative.org/api/bioentity/disease/MONDO:0100096/phenotypes?direct=True&rows=200&unselect_evidence=True\n",
      "API 9.1 BioLink API: 0 hits\n",
      "API 9.2: https://api.monarchinitiative.org/api/bioentity/disease/MONDO:0100096/genes?direct=True&rows=200&unselect_evidence=True\n",
      "API 9.2 BioLink API: 2 hits\n",
      "API 9.3: https://api.monarchinitiative.org/api/bioentity/disease/MONDO:0100096/genes?direct=True&rows=200&unselect_evidence=True\n",
      "API 9.3 BioLink API: 0 hits\n",
      "API 9.4: https://api.monarchinitiative.org/api/bioentity/disease/MONDO:0100096/pathways?direct=True&rows=200&unselect_evidence=True\n",
      "API 9.4 BioLink API: 0 hits\n",
      "API 8.1: https://pending.biothings.io/DISEASES/query?fields=DISEASES.associatedWith (POST -d q=DOID:0080600&scopes=DISEASES.doid)\n",
      "API 8.1 DISEASES API: 49 hits\n",
      "API 5.1: https://pending.biothings.io/mgigene2phenotype/query?fields=_id&size=300 (POST -d q=DOID:0080600&scopes=mgi.associated_with_disease.doid)\n",
      "API 5.1 MGIgene2phenotype API: 0 hits\n",
      "API 10.1: https://translator.broadinstitute.org/molepro_reasoner/query (POST -d {'message': {'query_graph': {'edges': [{'id': 'e00', 'source_id': 'n00', 'target_id': 'n01', 'type': 'related_to'}], 'nodes': [{'curie': 'MONDO:0100096', 'id': 'n00', 'type': 'disease'}, {'id': 'n01', 'type': 'gene'}]}}})\n",
      "API 10.1 Molecular Data Provider API: 0 hits\n",
      "API 7.1: https://automat.renci.org/pharos/disease/chemical_substance/MONDO:0100096\n",
      "API call to Automat PHAROS API with input MONDO:0100096 failed with status code 404\n",
      "{'error': 'Failed to get response from http://pharos-plater-plater-service:8080/disease/chemical_substance/MONDO:0100096?. Status code 404'}\n",
      "API 7.1: https://automat.renci.org/pharos/disease/chemical_substance/MONDO:0100096\n",
      "API call to Automat PHAROS API with input MONDO:0100096 failed with status code 404\n",
      "{'error': 'Failed to get response from http://pharos-plater-plater-service:8080/disease/chemical_substance/MONDO:0100096?. Status code 404'}\n",
      "API 2.1: https://automat.renci.org/cord19-scibite/disease/chemical_substance/MONDO:0100096\n",
      "API 2.1 Automat CORD19 Scibite API: 0 hits\n",
      "API 7.1: https://automat.renci.org/pharos/disease/gene/MONDO:0100096\n",
      "API 7.1 Automat PHAROS API: 0 hits\n",
      "API 11.1: https://automat.renci.org/hmdb/disease/chemical_substance/MONDO:0100096\n",
      "API 11.1 Automat HMDB API: 0 hits\n",
      "API 2.2: https://automat.renci.org/cord19-scibite/disease/gene/MONDO:0100096\n",
      "API 2.2 Automat CORD19 Scibite API: 0 hits\n",
      "API 2.3: https://automat.renci.org/cord19-scibite/disease/chemical_substance/MONDO:0100096\n",
      "API 2.3 Automat CORD19 Scibite API: 0 hits\n",
      "API 11.2: https://automat.renci.org/hmdb/disease/chemical_substance/MONDO:0100096\n",
      "API 11.2 Automat HMDB API: 0 hits\n",
      "API 4.1: https://automat.renci.org/cord19-scigraph/disease/gene/MONDO:0100096\n",
      "API 4.1 Automat CORD19 Scigraph API: 7 hits\n",
      "API 4.2: https://automat.renci.org/cord19-scigraph/disease/chemical_substance/MONDO:0100096\n",
      "API 4.2 Automat CORD19 Scigraph API: 13 hits\n",
      "API 4.3: https://automat.renci.org/cord19-scigraph/disease/chemical_substance/MONDO:0100096\n",
      "API 4.3 Automat CORD19 Scigraph API: 0 hits\n",
      "API 1.1: https://www.ebi.ac.uk/ols/api/ontologies/doid/children?id=DOID:0080600\n",
      "API 1.1 Ontology Lookup Service API: 0 hits\n",
      "API 12.5: https://biothings.ncats.io/cord_disease/query?fields=associated_with (POST -d q=DOID:0080600&scopes=doid)\n",
      "API 12.5 CORD Disease API: 0 hits\n",
      "API 12.6: https://biothings.ncats.io/cord_disease/query?fields=associated_with (POST -d q=DOID:0080600&scopes=doid)\n",
      "API 12.6 CORD Disease API: 0 hits\n",
      "API 12.7: https://biothings.ncats.io/cord_disease/query?fields=associated_with (POST -d q=DOID:0080600&scopes=doid)\n",
      "API 12.7 CORD Disease API: 0 hits\n",
      "API 12.8: https://biothings.ncats.io/cord_disease/query?fields=associated_with (POST -d q=DOID:0080600&scopes=doid)\n",
      "API 12.8 CORD Disease API: 0 hits\n",
      "\n",
      "BTE starts to perform id-to-object translation.\n",
      "\n",
      "After id-to-object translation, BTE retrieved 69 unique output nodes.\n",
      "\n",
      "\n",
      "\n",
      "After applying post-query filter, BTE retrieved 69 unique output nodes.\n",
      "\n",
      "========== Query # 2 ==========\n",
      "\n",
      "==== Step #1: Query Path Planning ====\n",
      "\n",
      "Input Types: Disease,Gene,ChemicalSubstance\n",
      "Output Types: Gene\n",
      "Predicates: None\n",
      "\n",
      "BTE found 21 APIs based on SmartAPI Meta-KG.\n",
      "\n",
      "API 1. Automat CORD19 Scibite API (17 API calls)\n",
      "API 2. Automat CORD19 Scigraph API (17 API calls)\n",
      "API 3. MGIgene2phenotype API (1 API calls)\n",
      "API 4. TCGA Mutation Frequency KP API (1 API calls)\n",
      "API 5. Automat PHAROS API (17 API calls)\n",
      "API 6. DISEASES API (1 API calls)\n",
      "API 7. BioLink API (110 API calls)\n",
      "API 8. Molecular Data Provider API (1 API calls)\n",
      "API 9. CORD Disease API (1 API calls)\n",
      "API 10. CORD Gene API (1 API calls)\n",
      "API 11. MyGene.info API (1 API calls)\n",
      "API 12. SEMMED Gene API (10 API calls)\n",
      "API 13. Text Mining Targeted Association API (32 API calls)\n",
      "API 14. SEMMED Chemical API (13 API calls)\n",
      "API 15. MyChem.info API (3 API calls)\n",
      "API 16. CTD API (10 API calls)\n",
      "API 17. CORD Chemical API (1 API calls)\n",
      "API 18. Automat CHEMBIO API (16 API calls)\n",
      "API 19. Automat HMDB API (16 API calls)\n",
      "API 20. Drug Response KP API (9 API calls)\n",
      "API 21. BioThings DGIdb API (1 API calls)\n",
      "\n",
      "\n",
      "==== Step #2: Query path execution ====\n",
      "\n",
      "NOTE: API requests are dispatched in parallel, so the list of APIs below is ordered by query time.\n",
      "\n",
      "API 7.1: https://api.monarchinitiative.org/api/bioentity/disease/MONDO:0020753/genes?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.1 BioLink API: 0 hits\n",
      "API 7.2: https://api.monarchinitiative.org/api/bioentity/disease/MONDO:0020753/genes?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.2 BioLink API: 0 hits\n",
      "API 15.1: https://mychem.info/v1/query?fields=drugcentral.bioactivity (POST -d q=CHEMBL283800,CHEMBL47047,CHEMBL1535,CHEMBL58510,CHEMBL76,CHEMBL1326,CHEMBL729,CHEMBL163,CHEMBL2095223,CHEMBL1669,CHEMBL1214598,CHEMBL1234886,CHEMBL1690&scopes=chembl.molecule_chembl_id)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API 15.1 MyChem.info API: 31 hits\n",
      "API 16.1: http://ctdbase.org/tools/batchQuery.go?format=json&inputTerms=D019438&inputType=chem&report=genes_curated\n",
      "API 4.1: https://biothings.ncats.io/tcga_mut_freq_kp/query?fields=association.freq_by_sample,subject.SYMBOL&q=object.id:\"MONDO:0020753\"%20AND%20association.freq_by_sample:>0.1&size=1000&sort=-association.freq_by_sample\n",
      "API 4.1 TCGA Mutation Frequency KP API: 0 hits\n",
      "API 3.1: https://pending.biothings.io/mgigene2phenotype/query?fields=_id&size=300 (POST -d q=DOID:0080599&scopes=mgi.associated_with_disease.doid)\n",
      "API 3.1 MGIgene2phenotype API: 0 hits\n",
      "API 6.1: https://pending.biothings.io/DISEASES/query?fields=DISEASES.associatedWith (POST -d q=DOID:0080599&scopes=DISEASES.doid)\n",
      "API 6.1 DISEASES API: 0 hits\n",
      "API 9.1: https://biothings.ncats.io/cord_disease/query?fields=associated_with (POST -d q=DOID:0080599&scopes=doid)\n",
      "API call to CORD Disease API with input ['DOID:0080599'] failed with unknown response\n",
      "API 15.2: https://mychem.info/v1/query?fields=drugbank.targets (POST -d q=DB00608,DB01601,DB09140,DB01611,DB13609,DB00503,DB00148&scopes=drugbank.id)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 297, in callSingleAPI\n",
      "    result = tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/__init__.py\", line 49, in transform\n",
      "    return self.tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/base_transformer.py\", line 78, in transform\n",
      "    item = self.wrap(item)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/cord_transformer.py\", line 16, in wrap\n",
      "    for k in item.keys():\n",
      "RuntimeError: dictionary keys changed during iteration\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API 15.2 MyChem.info API: 21 hits\n",
      "API 16.1 CTD API: 127 hits\n",
      "API 1.1: https://automat.renci.org/cord19-scibite/disease/gene/MONDO:0020753\n",
      "API 1.1 Automat CORD19 Scibite API: 0 hits\n",
      "API 19.1: https://automat.renci.org/hmdb/chemical_substance/gene/CHEBI:33098\n",
      "API 19.1 Automat HMDB API: 0 hits\n",
      "API 1.2: https://automat.renci.org/cord19-scibite/chemical_substance/gene/CHEBI:134730\n",
      "API 1.2 Automat CORD19 Scibite API: 0 hits\n",
      "API 18.1: https://automat.renci.org/chembio/chemical_substance/gene/CHEBI:45409\n",
      "API 18.1 Automat CHEMBIO API: 0 hits\n",
      "API 2.1: https://automat.renci.org/cord19-scigraph/chemical_substance/gene/CHEBI:5801\n",
      "API 2.1 Automat CORD19 Scigraph API: 2 hits\n",
      "API 2.2: https://automat.renci.org/cord19-scigraph/chemical_substance/gene/CHEBI:134730\n",
      "API 2.2 Automat CORD19 Scigraph API: 0 hits\n",
      "API 2.3: https://automat.renci.org/cord19-scigraph/disease/gene/MONDO:0020753\n",
      "API 2.3 Automat CORD19 Scigraph API: 0 hits\n",
      "API 5.1: https://automat.renci.org/pharos/chemical_substance/gene/CHEBI:134730\n",
      "API 5.1 Automat PHAROS API: 0 hits\n",
      "API 18.2: https://automat.renci.org/chembio/chemical_substance/gene/CHEBI:33098\n",
      "API 18.2 Automat CHEMBIO API: 0 hits\n",
      "API 19.2: https://automat.renci.org/hmdb/chemical_substance/gene/CHEBI:134730\n",
      "API 19.2 Automat HMDB API: 0 hits\n",
      "API 5.2: https://automat.renci.org/pharos/chemical_substance/gene/CHEBI:33098\n",
      "API 5.2 Automat PHAROS API: 0 hits\n",
      "API 1.3: https://automat.renci.org/cord19-scibite/chemical_substance/gene/CHEBI:5801\n",
      "API 1.3 Automat CORD19 Scibite API: 0 hits\n",
      "API 13.1: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:45409\" AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 13.1 Text Mining Targeted Association API: 0 hits\n",
      "API 20.1: https://biothings.ncats.io/drug_response_kp/query?fields=association.context.disease.mondo,subject.NCBIGene,association.effect_size,association.pvalue&q=object.PUBCHEM:3652%20AND%20association.effect_size:<0%20AND%20association.pvalue:<0.05&size=1000&sort=association.pvalue\n",
      "API 20.1 Drug Response KP API: 0 hits\n",
      "API 13.2: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:134730\" AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 13.2 Text Mining Targeted Association API: 0 hits\n",
      "API 20.2: https://biothings.ncats.io/drug_response_kp/query?fields=association.context.disease.mondo,subject.NCBIGene,association.effect_size,association.pvalue&q=object.PUBCHEM:28446253%20AND%20association.effect_size:<0%20AND%20association.pvalue:<0.05&size=1000&sort=association.pvalue\n",
      "API 20.2 Drug Response KP API: 0 hits\n",
      "API 13.3: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:33098\" AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 13.3 Text Mining Targeted Association API: 0 hits\n",
      "API 20.3: https://biothings.ncats.io/drug_response_kp/query?fields=association.context.disease.mondo,subject.NCBIGene,association.effect_size,association.pvalue&q=object.PUBCHEM:131411%20AND%20association.effect_size:<0%20AND%20association.pvalue:<0.05&size=1000&sort=association.pvalue\n",
      "API 20.3 Drug Response KP API: 0 hits\n",
      "API 15.3: https://mychem.info/v1/query?fields=drugbank.enzymes (POST -d q=DB00608,DB01601,DB09140,DB01611,DB13609,DB00503,DB00148&scopes=drugbank.id)\n",
      "API 15.3 MyChem.info API: 33 hits\n",
      "API 11.1: https://mygene.info/v3/query?fields=pantherdb.ortholog (POST -d q=92521,5499,7124,3439,1506,3574,10044,925,2580,28,2147,3558,10410,10473,5045,3605,59272,7113,3627,7137,2875,3576,4151,3559,5447,3586,1514,3553,10045,6868,3569,6651,6347,10332,682,6348,3456,54585,3458,22848,213,920,7054,54938,186,1636,4141,5972,339967,2152,1401,185,1803,340024&scopes=entrezgene)\n",
      "API 11.1 MyGene.info API: 82 hits\n",
      "API 16.2: http://ctdbase.org/tools/batchQuery.go?format=json&inputTerms=C086979&inputType=chem&report=genes_curated\n",
      "API 16.2 CTD API: 0 hits\n",
      "API 13.4: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:5801\" AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 13.4 Text Mining Targeted Association API: 11 hits\n",
      "API 20.4: https://biothings.ncats.io/drug_response_kp/query?fields=association.context.disease.mondo,subject.NCBIGene,association.effect_size,association.pvalue&q=object.PUBCHEM:6944%20AND%20association.effect_size:<0%20AND%20association.pvalue:<0.05&size=1000&sort=association.pvalue\n",
      "API 20.4 Drug Response KP API: 0 hits\n",
      "API 16.3: http://ctdbase.org/tools/batchQuery.go?format=json&inputTerms=C029955&inputType=chem&report=genes_curated\n",
      "API 16.3 CTD API: 67 hits\n",
      "API 21.1: https://biothings.ncats.io/dgidb/query?fields=subject.NCBIGene,association.provided_by,association.pubmed&size=1000 (POST -d q=CHEMBL283800,CHEMBL47047,CHEMBL1535,CHEMBL58510,CHEMBL76,CHEMBL1326,CHEMBL729,CHEMBL163,CHEMBL2095223,CHEMBL1669,CHEMBL1214598,CHEMBL1234886,CHEMBL1690&scopes=object.CHEMBL_COMPOUND)\n",
      "API 21.1 BioThings DGIdb API: 25 hits\n",
      "API 16.4: http://ctdbase.org/tools/batchQuery.go?format=json&inputTerms=D006886&inputType=chem&report=genes_curated\n",
      "API 16.4 CTD API: 81 hits\n",
      "API 14.1: https://biothings.ncats.io/semmedchemical/query?fields=affects (POST -d q=C0020336,C0254211,C0046462,C0030054,C0008269,C0700447,C0003751,C0010286,C0292818,C0674432&scopes=umls)\n",
      "API 14.1 SEMMED Chemical API: 0 hits\n",
      "API 7.3: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7113/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.3 BioLink API: 71 hits\n",
      "API 17.1: https://biothings.ncats.io/cord_chemical/query?fields=associated_with (POST -d q=CHEBI:132943,CHEBI:5801,CHEBI:25805,CHEBI:33098,CHEBI:134730,CHEBI:3638,CHEBI:45409,CHEBI:15379,CHEBI:16919,CHEBI:29194,CHEBI:24261,CHEBI:31781,CHEBI:27140,CHEBI:57947,CHEBI:48433,CHEBI:29995&scopes=chebi)\n",
      "API 14.2: https://biothings.ncats.io/semmedchemical/query?fields=disrupted_by (POST -d q=C0020336,C0254211,C0046462,C0030054,C0008269,C0700447,C0003751,C0010286,C0292818,C0674432&scopes=umls)\n",
      "API 14.2 SEMMED Chemical API: 0 hits\n",
      "API 14.3: https://biothings.ncats.io/semmedchemical/query?fields=coexists_with (POST -d q=C0020336,C0254211,C0046462,C0030054,C0008269,C0700447,C0003751,C0010286,C0292818,C0674432&scopes=umls)\n",
      "API 14.3: https://biothings.ncats.io/semmedchemical/query?fields=affected_by (POST -d q=C0020336,C0254211,C0046462,C0030054,C0008269,C0700447,C0003751,C0010286,C0292818,C0674432&scopes=umls)\n",
      "API 14.3 SEMMED Chemical API: 0 hits\n",
      "API 12.1: https://biothings.ncats.io/semmedgene/query?fields=disrupts (POST -d q=C1415900,C1439284,C1428660,C1422064,C1414944,C1825592,C1420826,C1414141,C1415898,C1823096,C1334124,C1333522,C1413931,C1335240,C1415912,C1539099,C1417048,C1333916,C1337092,C1412099,C1416946,C1420310,C1332690,C0081714,C1540024,C1413244,C1417035,C1425041,C0054871,C1413716,C1415274,C1412186,C1426329,C1335818,C1335817,C1419338,C0035094,C1334122,C1420718,C1366571,C0812246,C1412828,C1332036,C1334112,C1367714,C1412332,C1334085,C0021745,C1332807,C1332714,C1413809,C1334098,C1414504,C1335232,C1336641,C0919550,C1334114,C0879590&scopes=umls)\n",
      "API 12.1: https://biothings.ncats.io/semmedgene/query?fields=disrupted_by (POST -d q=C1415900,C1439284,C1428660,C1422064,C1414944,C1825592,C1420826,C1414141,C1415898,C1823096,C1334124,C1333522,C1413931,C1335240,C1415912,C1539099,C1417048,C1333916,C1337092,C1412099,C1416946,C1420310,C1332690,C0081714,C1540024,C1413244,C1417035,C1425041,C0054871,C1413716,C1415274,C1412186,C1426329,C1335818,C1335817,C1419338,C0035094,C1334122,C1420718,C1366571,C0812246,C1412828,C1332036,C1334112,C1367714,C1412332,C1334085,C0021745,C1332807,C1332714,C1413809,C1334098,C1414504,C1335232,C1336641,C0919550,C1334114,C0879590&scopes=umls)\n",
      "API 12.1 SEMMED Gene API: 0 hits\n",
      "API 12.2: https://biothings.ncats.io/semmedgene/query?fields=affects (POST -d q=C1415900,C1439284,C1428660,C1422064,C1414944,C1825592,C1420826,C1414141,C1415898,C1823096,C1334124,C1333522,C1413931,C1335240,C1415912,C1539099,C1417048,C1333916,C1337092,C1412099,C1416946,C1420310,C1332690,C0081714,C1540024,C1413244,C1417035,C1425041,C0054871,C1413716,C1415274,C1412186,C1426329,C1335818,C1335817,C1419338,C0035094,C1334122,C1420718,C1366571,C0812246,C1412828,C1332036,C1334112,C1367714,C1412332,C1334085,C0021745,C1332807,C1332714,C1413809,C1334098,C1414504,C1335232,C1336641,C0919550,C1334114,C0879590&scopes=umls)\n",
      "API 5.3: https://automat.renci.org/pharos/chemical_substance/gene/CHEBI:5801\n",
      "API 5.3 Automat PHAROS API: 2 hits\n",
      "API 2.4: https://automat.renci.org/cord19-scigraph/chemical_substance/gene/CHEBI:33098\n",
      "API 2.4 Automat CORD19 Scigraph API: 16 hits\n",
      "API 1.4: https://automat.renci.org/cord19-scibite/chemical_substance/gene/CHEBI:33098\n",
      "API 1.4 Automat CORD19 Scibite API: 0 hits\n",
      "API 18.3: https://automat.renci.org/chembio/chemical_substance/gene/CHEBI:5801\n",
      "API 18.3 Automat CHEMBIO API: 2 hits\n",
      "API 19.3: https://automat.renci.org/hmdb/chemical_substance/gene/CHEBI:5801\n",
      "API 19.3 Automat HMDB API: 2 hits\n",
      "API 19.4: https://automat.renci.org/hmdb/chemical_substance/gene/CHEBI:45409\n",
      "API 19.4 Automat HMDB API: 16 hits\n",
      "API 18.4: https://automat.renci.org/chembio/chemical_substance/gene/CHEBI:134730\n",
      "API 18.4 Automat CHEMBIO API: 0 hits\n",
      "API 5.4: https://automat.renci.org/pharos/disease/gene/MONDO:0020753\n",
      "API 5.4 Automat PHAROS API: 0 hits\n",
      "API 8.1: https://translator.broadinstitute.org/molepro_reasoner/query (POST -d {'message': {'query_graph': {'edges': [{'id': 'e00', 'source_id': 'n00', 'target_id': 'n01', 'type': 'related_to'}], 'nodes': [{'curie': 'MONDO:0020753', 'id': 'n00', 'type': 'disease'}, {'id': 'n01', 'type': 'gene'}]}}})\n",
      "API 8.1 Molecular Data Provider API: 0 hits\n",
      "API 10.1: https://biothings.ncats.io/cord_gene/query?fields=associated_with (POST -d q=11892,9281,19679,79,2707,2367,10618,399,27954,6915,5438,1116,6018,5414,9208,6898,13523,8568,11183,10637,16884,30615,3541,6741,13557,1706,6008,4552,11782,4989,5981,11876,11947,17697,10627,6023,338,5992,1678,6001,27960,336,5417,6025,16885,9958,4113,5962,195,3535,2537,5434,2524,3009&scopes=hgnc)\n",
      "API call to CORD Chemical API with input ['CHEBI:132943', 'CHEBI:5801', 'CHEBI:25805', 'CHEBI:33098', 'CHEBI:134730', 'CHEBI:3638', 'CHEBI:45409', 'CHEBI:15379', 'CHEBI:16919', 'CHEBI:29194', 'CHEBI:24261', 'CHEBI:31781', 'CHEBI:27140', 'CHEBI:57947', 'CHEBI:48433', 'CHEBI:29995'] failed with unknown response\n",
      "API 14.4 SEMMED Chemical API: 710 hits\n",
      "API 12.2 SEMMED Gene API: 1 hits\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 297, in callSingleAPI\n",
      "    result = tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/__init__.py\", line 49, in transform\n",
      "    return self.tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/base_transformer.py\", line 78, in transform\n",
      "    item = self.wrap(item)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/cord_transformer.py\", line 16, in wrap\n",
      "    for k in item.keys():\n",
      "RuntimeError: dictionary keys changed during iteration\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API 12.3 SEMMED Gene API: 2 hits\n",
      "API 12.4: https://biothings.ncats.io/semmedgene/query?fields=affected_by (POST -d q=C1415900,C1439284,C1428660,C1422064,C1414944,C1825592,C1420826,C1414141,C1415898,C1823096,C1334124,C1333522,C1413931,C1335240,C1415912,C1539099,C1417048,C1333916,C1337092,C1412099,C1416946,C1420310,C1332690,C0081714,C1540024,C1413244,C1417035,C1425041,C0054871,C1413716,C1415274,C1412186,C1426329,C1335818,C1335817,C1419338,C0035094,C1334122,C1420718,C1366571,C0812246,C1412828,C1332036,C1334112,C1367714,C1412332,C1334085,C0021745,C1332807,C1332714,C1413809,C1334098,C1414504,C1335232,C1336641,C0919550,C1334114,C0879590&scopes=umls)\n",
      "API 7.4: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:59272/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 12.4 SEMMED Gene API: 4 hits\n",
      "API 7.4 BioLink API: 28 hits\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 297, in callSingleAPI\n",
      "    result = tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/__init__.py\", line 49, in transform\n",
      "    return self.tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/base_transformer.py\", line 78, in transform\n",
      "    item = self.wrap(item)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/cord_transformer.py\", line 16, in wrap\n",
      "    for k in item.keys():\n",
      "RuntimeError: dictionary keys changed during iteration\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API call to CORD Gene API with input ['11892', '9281', '19679', '79', '2707', '2367', '10618', '399', '27954', '6915', '5438', '1116', '6018', '5414', '9208', '6898', '13523', '8568', '11183', '10637', '16884', '30615', '3541', '6741', '13557', '1706', '6008', '4552', '11782', '4989', '5981', '11876', '11947', '17697', '10627', '6023', '338', '5992', '1678', '6001', '27960', '336', '5417', '6025', '16885', '9958', '4113', '5962', '195', '3535', '2537', '5434', '2524', '3009'] failed with unknown response\n",
      "API 13.5: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:3638\" AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 13.5 Text Mining Targeted Association API: 54 hits\n",
      "API 13.6: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:16919\" AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 13.6 Text Mining Targeted Association API: 8 hits\n",
      "API 13.7: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:15379\" AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 13.7 Text Mining Targeted Association API: 0 hits\n",
      "API 13.8: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:27140\" AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 13.8 Text Mining Targeted Association API: 0 hits\n",
      "API 20.5: https://biothings.ncats.io/drug_response_kp/query?fields=association.context.disease.mondo,subject.NCBIGene,association.effect_size,association.pvalue&q=object.PUBCHEM:977%20AND%20association.effect_size:<0%20AND%20association.pvalue:<0.05&size=1000&sort=association.pvalue\n",
      "API 20.5 Drug Response KP API: 0 hits\n",
      "API 20.6: https://biothings.ncats.io/drug_response_kp/query?fields=association.context.disease.mondo,subject.NCBIGene,association.effect_size,association.pvalue&q=object.PUBCHEM:586%20AND%20association.effect_size:<0%20AND%20association.pvalue:<0.05&size=1000&sort=association.pvalue\n",
      "API 20.6 Drug Response KP API: 0 hits\n",
      "API 20.7: https://biothings.ncats.io/drug_response_kp/query?fields=association.context.disease.mondo,subject.NCBIGene,association.effect_size,association.pvalue&q=object.PUBCHEM:2719%20AND%20association.effect_size:<0%20AND%20association.pvalue:<0.05&size=1000&sort=association.pvalue\n",
      "API 20.7 Drug Response KP API: 0 hits\n",
      "API 20.8: https://biothings.ncats.io/drug_response_kp/query?fields=association.context.disease.mondo,subject.NCBIGene,association.effect_size,association.pvalue&q=object.PUBCHEM:392622%20AND%20association.effect_size:<0%20AND%20association.pvalue:<0.05&size=1000&sort=association.pvalue\n",
      "API 20.8 Drug Response KP API: 0 hits\n",
      "API 20.9: https://biothings.ncats.io/drug_response_kp/query?fields=association.context.disease.mondo,subject.NCBIGene,association.effect_size,association.pvalue&q=object.PUBCHEM:92727%20AND%20association.effect_size:<0%20AND%20association.pvalue:<0.05&size=1000&sort=association.pvalue\n",
      "API 20.9 Drug Response KP API: 0 hits\n",
      "API 1.5: https://automat.renci.org/cord19-scibite/chemical_substance/gene/CHEBI:25805\n",
      "API 1.5 Automat CORD19 Scibite API: 0 hits\n",
      "API 1.6: https://automat.renci.org/cord19-scibite/chemical_substance/gene/CHEBI:27140\n",
      "API 1.6 Automat CORD19 Scibite API: 0 hits\n",
      "API 1.7: https://automat.renci.org/cord19-scibite/chemical_substance/gene/CHEBI:15379\n",
      "API 1.7 Automat CORD19 Scibite API: 0 hits\n",
      "API 1.8: https://automat.renci.org/cord19-scibite/chemical_substance/gene/CHEBI:3638\n",
      "API 1.8 Automat CORD19 Scibite API: 2 hits\n",
      "API 18.5: https://automat.renci.org/chembio/chemical_substance/gene/CHEBI:27140\n",
      "API 18.5 Automat CHEMBIO API: 0 hits\n",
      "API 5.5: https://automat.renci.org/pharos/chemical_substance/gene/CHEBI:25805\n",
      "API 5.5 Automat PHAROS API: 0 hits\n",
      "API 18.6: https://automat.renci.org/chembio/chemical_substance/gene/CHEBI:15379\n",
      "API 18.6 Automat CHEMBIO API: 0 hits\n",
      "API 18.7: https://automat.renci.org/chembio/chemical_substance/gene/CHEBI:25805\n",
      "API 18.7 Automat CHEMBIO API: 0 hits\n",
      "API 5.6: https://automat.renci.org/pharos/chemical_substance/gene/CHEBI:15379\n",
      "API 5.6 Automat PHAROS API: 0 hits\n",
      "API 18.8: https://automat.renci.org/chembio/chemical_substance/gene/CHEBI:3638\n",
      "API 18.8 Automat CHEMBIO API: 0 hits\n",
      "API 5.7: https://automat.renci.org/pharos/chemical_substance/gene/CHEBI:27140\n",
      "API 5.7 Automat PHAROS API: 0 hits\n",
      "API 18.9: https://automat.renci.org/chembio/chemical_substance/gene/CHEBI:16919\n",
      "API 18.9 Automat CHEMBIO API: 0 hits\n",
      "API 5.8: https://automat.renci.org/pharos/chemical_substance/gene/CHEBI:45409\n",
      "API 5.8 Automat PHAROS API: 4 hits\n",
      "API 2.5: https://automat.renci.org/cord19-scigraph/chemical_substance/gene/CHEBI:3638\n",
      "API 2.5 Automat CORD19 Scigraph API: 9 hits\n",
      "API 2.6: https://automat.renci.org/cord19-scigraph/chemical_substance/gene/CHEBI:25805\n",
      "API 2.6 Automat CORD19 Scigraph API: 0 hits\n",
      "API 5.9: https://automat.renci.org/pharos/chemical_substance/gene/CHEBI:3638\n",
      "API 5.9 Automat PHAROS API: 2 hits\n",
      "API 2.7: https://automat.renci.org/cord19-scigraph/chemical_substance/gene/CHEBI:45409\n",
      "API 2.7 Automat CORD19 Scigraph API: 5 hits\n",
      "API 2.8: https://automat.renci.org/cord19-scigraph/chemical_substance/gene/CHEBI:27140\n",
      "API 2.8 Automat CORD19 Scigraph API: 0 hits\n",
      "API 2.9: https://automat.renci.org/cord19-scigraph/chemical_substance/gene/CHEBI:15379\n",
      "API 2.9 Automat CORD19 Scigraph API: 27 hits\n",
      "API 13.9: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:25805\" AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 16.5: http://ctdbase.org/tools/batchQuery.go?format=json&inputTerms=D003401&inputType=chem&report=genes_curated\n",
      "API 16.5 CTD API: 14 hits\n",
      "API 16.6: http://ctdbase.org/tools/batchQuery.go?format=json&inputTerms=D002738&inputType=chem&report=genes_curated\n",
      "API 16.6: http://ctdbase.org/tools/batchQuery.go?format=json&inputTerms=D061466&inputType=chem&report=genes_curated\n",
      "API 16.6 CTD API: 16 hits\n",
      "API 13.9 Text Mining Targeted Association API: 199 hits\n",
      "API 16.7 CTD API: 324 hits\n",
      "API 14.5: https://biothings.ncats.io/semmedchemical/query?fields=physically_interacts_with (POST -d q=C0020336,C0254211,C0046462,C0030054,C0008269,C0700447,C0003751,C0010286,C0292818,C0674432&scopes=umls)\n",
      "API 16.8: http://ctdbase.org/tools/batchQuery.go?format=json&inputTerms=C023676&inputType=chem&report=genes_curated\n",
      "API 16.8 CTD API: 65 hits\n",
      "API 7.5: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:1401/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.5 BioLink API: 120 hits\n",
      "API 14.5: https://biothings.ncats.io/semmedchemical/query?fields=positively_regulated_by (POST -d q=C0020336,C0254211,C0046462,C0030054,C0008269,C0700447,C0003751,C0010286,C0292818,C0674432&scopes=umls)\n",
      "API 14.5 SEMMED Chemical API: 299 hits\n",
      "API 16.9: http://ctdbase.org/tools/batchQuery.go?format=json&inputTerms=D010100&inputType=chem&report=genes_curated\n",
      "API 14.6: https://biothings.ncats.io/semmedchemical/query?fields=negatively_regulated_by (POST -d q=C0020336,C0254211,C0046462,C0030054,C0008269,C0700447,C0003751,C0010286,C0292818,C0674432&scopes=umls)\n",
      "API 14.6 SEMMED Chemical API: 212 hits\n",
      "API 14.7: https://biothings.ncats.io/semmedchemical/query?fields=negatively_regulates (POST -d q=C0020336,C0254211,C0046462,C0030054,C0008269,C0700447,C0003751,C0010286,C0292818,C0674432&scopes=umls)\n",
      "API 14.7 SEMMED Chemical API: 564 hits\n",
      "API 14.8: https://biothings.ncats.io/semmedchemical/query?fields=disrupts (POST -d q=C0020336,C0254211,C0046462,C0030054,C0008269,C0700447,C0003751,C0010286,C0292818,C0674432&scopes=umls)\n",
      "API 14.8 SEMMED Chemical API: 0 hits\n",
      "API 7.6: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:5447/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.6 BioLink API: 137 hits\n",
      "API 12.5: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulates (POST -d q=C1415900,C1439284,C1428660,C1422064,C1414944,C1825592,C1420826,C1414141,C1415898,C1823096,C1334124,C1333522,C1413931,C1335240,C1415912,C1539099,C1417048,C1333916,C1337092,C1412099,C1416946,C1420310,C1332690,C0081714,C1540024,C1413244,C1417035,C1425041,C0054871,C1413716,C1415274,C1412186,C1426329,C1335818,C1335817,C1419338,C0035094,C1334122,C1420718,C1366571,C0812246,C1412828,C1332036,C1334112,C1367714,C1412332,C1334085,C0021745,C1332807,C1332714,C1413809,C1334098,C1414504,C1335232,C1336641,C0919550,C1334114,C0879590&scopes=umls)\n",
      "API 1.9: https://automat.renci.org/cord19-scibite/chemical_substance/gene/CHEBI:45409\n",
      "API 1.9 Automat CORD19 Scibite API: 7 hits\n",
      "API 12.5: https://biothings.ncats.io/semmedgene/query?fields=positively_regulated_by (POST -d q=C1415900,C1439284,C1428660,C1422064,C1414944,C1825592,C1420826,C1414141,C1415898,C1823096,C1334124,C1333522,C1413931,C1335240,C1415912,C1539099,C1417048,C1333916,C1337092,C1412099,C1416946,C1420310,C1332690,C0081714,C1540024,C1413244,C1417035,C1425041,C0054871,C1413716,C1415274,C1412186,C1426329,C1335818,C1335817,C1419338,C0035094,C1334122,C1420718,C1366571,C0812246,C1412828,C1332036,C1334112,C1367714,C1412332,C1334085,C0021745,C1332807,C1332714,C1413809,C1334098,C1414504,C1335232,C1336641,C0919550,C1334114,C0879590&scopes=umls)\n",
      "API 12.5: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulated_by (POST -d q=C1415900,C1439284,C1428660,C1422064,C1414944,C1825592,C1420826,C1414141,C1415898,C1823096,C1334124,C1333522,C1413931,C1335240,C1415912,C1539099,C1417048,C1333916,C1337092,C1412099,C1416946,C1420310,C1332690,C0081714,C1540024,C1413244,C1417035,C1425041,C0054871,C1413716,C1415274,C1412186,C1426329,C1335818,C1335817,C1419338,C0035094,C1334122,C1420718,C1366571,C0812246,C1412828,C1332036,C1334112,C1367714,C1412332,C1334085,C0021745,C1332807,C1332714,C1413809,C1334098,C1414504,C1335232,C1336641,C0919550,C1334114,C0879590&scopes=umls)\n",
      "API 7.7: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7054/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.7: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:4141/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.7: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:6651/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 14.9 SEMMED Chemical API: 971 hits\n",
      "API 12.5: https://biothings.ncats.io/semmedgene/query?fields=physically_interacts_with (POST -d q=C1415900,C1439284,C1428660,C1422064,C1414944,C1825592,C1420826,C1414141,C1415898,C1823096,C1334124,C1333522,C1413931,C1335240,C1415912,C1539099,C1417048,C1333916,C1337092,C1412099,C1416946,C1420310,C1332690,C0081714,C1540024,C1413244,C1417035,C1425041,C0054871,C1413716,C1415274,C1412186,C1426329,C1335818,C1335817,C1419338,C0035094,C1334122,C1420718,C1366571,C0812246,C1412828,C1332036,C1334112,C1367714,C1412332,C1334085,C0021745,C1332807,C1332714,C1413809,C1334098,C1414504,C1335232,C1336641,C0919550,C1334114,C0879590&scopes=umls)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API 16.9 CTD API: 4588 hits\n",
      "API 12.5: https://biothings.ncats.io/semmedgene/query?fields=positively_regulates (POST -d q=C1415900,C1439284,C1428660,C1422064,C1414944,C1825592,C1420826,C1414141,C1415898,C1823096,C1334124,C1333522,C1413931,C1335240,C1415912,C1539099,C1417048,C1333916,C1337092,C1412099,C1416946,C1420310,C1332690,C0081714,C1540024,C1413244,C1417035,C1425041,C0054871,C1413716,C1415274,C1412186,C1426329,C1335818,C1335817,C1419338,C0035094,C1334122,C1420718,C1366571,C0812246,C1412828,C1332036,C1334112,C1367714,C1412332,C1334085,C0021745,C1332807,C1332714,C1413809,C1334098,C1414504,C1335232,C1336641,C0919550,C1334114,C0879590&scopes=umls)\n",
      "API 12.5 SEMMED Gene API: 2795 hits\n",
      "API 7.7 BioLink API: 96 hits\n",
      "API 19.5: https://automat.renci.org/hmdb/chemical_substance/gene/CHEBI:3638\n",
      "API 19.5 Automat HMDB API: 9 hits\n",
      "API 19.6: https://automat.renci.org/hmdb/chemical_substance/gene/CHEBI:25805\n",
      "API 19.6 Automat HMDB API: 0 hits\n",
      "API 19.7: https://automat.renci.org/hmdb/chemical_substance/gene/CHEBI:16919\n",
      "API 19.7 Automat HMDB API: 0 hits\n",
      "API 19.8: https://automat.renci.org/hmdb/chemical_substance/gene/CHEBI:15379\n",
      "API 19.8 Automat HMDB API: 197 hits\n",
      "API 19.9: https://automat.renci.org/hmdb/chemical_substance/gene/CHEBI:27140\n",
      "API 19.9 Automat HMDB API: 0 hits\n",
      "API 7.8 BioLink API: 103 hits\n",
      "API 7.9 BioLink API: 195 hits\n",
      "API 12.6 SEMMED Gene API: 4459 hits\n",
      "API 12.7 SEMMED Gene API: 3257 hits\n",
      "API 12.8 SEMMED Gene API: 6219 hits\n",
      "API 12.9 SEMMED Gene API: 4692 hits\n",
      "API 13.10: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:57947\" AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 13.10 Text Mining Targeted Association API: 0 hits\n",
      "API 13.11: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:29194\" AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 13.11 Text Mining Targeted Association API: 0 hits\n",
      "API 13.12: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:31781\" AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 13.12 Text Mining Targeted Association API: 0 hits\n",
      "API 13.13: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:29995\" AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 13.13 Text Mining Targeted Association API: 13 hits\n",
      "API 13.14: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:24261\" AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 14.10: https://biothings.ncats.io/semmedchemical/query?fields=related_to (POST -d q=C0020336,C0254211,C0046462,C0030054,C0008269,C0700447,C0003751,C0010286,C0292818,C0674432&scopes=umls)\n",
      "API 14.10: https://biothings.ncats.io/semmedchemical/query?fields=produces (POST -d q=C0020336,C0254211,C0046462,C0030054,C0008269,C0700447,C0003751,C0010286,C0292818,C0674432&scopes=umls)\n",
      "API 14.10 SEMMED Chemical API: 0 hits\n",
      "API 13.14 Text Mining Targeted Association API: 136 hits\n",
      "API 18.10: https://automat.renci.org/chembio/chemical_substance/gene/CHEBI:24261\n",
      "API 18.10 Automat CHEMBIO API: 0 hits\n",
      "API 5.10: https://automat.renci.org/pharos/chemical_substance/gene/CHEBI:57947\n",
      "API 5.10 Automat PHAROS API: 0 hits\n",
      "API 1.10: https://automat.renci.org/cord19-scibite/chemical_substance/gene/CHEBI:29995\n",
      "API 1.10 Automat CORD19 Scibite API: 0 hits\n",
      "API 1.11: https://automat.renci.org/cord19-scibite/chemical_substance/gene/CHEBI:57947\n",
      "API 1.11 Automat CORD19 Scibite API: 0 hits\n",
      "API 18.11: https://automat.renci.org/chembio/chemical_substance/gene/CHEBI:31781\n",
      "API 18.11 Automat CHEMBIO API: 0 hits\n",
      "API 18.12: https://automat.renci.org/chembio/chemical_substance/gene/CHEBI:57947\n",
      "API 18.12 Automat CHEMBIO API: 0 hits\n",
      "API 1.12: https://automat.renci.org/cord19-scibite/chemical_substance/gene/CHEBI:29194\n",
      "API 1.12 Automat CORD19 Scibite API: 0 hits\n",
      "API 18.13: https://automat.renci.org/chembio/chemical_substance/gene/CHEBI:29194\n",
      "API 18.13 Automat CHEMBIO API: 0 hits\n",
      "API 1.13: https://automat.renci.org/cord19-scibite/chemical_substance/gene/CHEBI:16919\n",
      "API 1.13 Automat CORD19 Scibite API: 0 hits\n",
      "API 1.14: https://automat.renci.org/cord19-scibite/chemical_substance/gene/CHEBI:24261\n",
      "API 1.14 Automat CORD19 Scibite API: 0 hits\n",
      "API 5.11: https://automat.renci.org/pharos/chemical_substance/gene/CHEBI:29194\n",
      "API 5.11 Automat PHAROS API: 0 hits\n",
      "API 5.12: https://automat.renci.org/pharos/chemical_substance/gene/CHEBI:16919\n",
      "API 5.12 Automat PHAROS API: 0 hits\n",
      "API 18.14: https://automat.renci.org/chembio/chemical_substance/gene/CHEBI:29995\n",
      "API 18.14 Automat CHEMBIO API: 0 hits\n",
      "API 19.10: https://automat.renci.org/hmdb/chemical_substance/gene/CHEBI:24261\n",
      "API 19.10 Automat HMDB API: 0 hits\n",
      "API 5.13: https://automat.renci.org/pharos/chemical_substance/gene/CHEBI:29995\n",
      "API 5.13 Automat PHAROS API: 0 hits\n",
      "API 5.14: https://automat.renci.org/pharos/chemical_substance/gene/CHEBI:24261\n",
      "API 5.14 Automat PHAROS API: 0 hits\n",
      "API 19.11: https://automat.renci.org/hmdb/chemical_substance/gene/CHEBI:57947\n",
      "API 19.11 Automat HMDB API: 10 hits\n",
      "API 19.12: https://automat.renci.org/hmdb/chemical_substance/gene/CHEBI:29194\n",
      "API 19.12 Automat HMDB API: 0 hits\n",
      "API 19.13: https://automat.renci.org/hmdb/chemical_substance/gene/CHEBI:31781\n",
      "API 19.13 Automat HMDB API: 7 hits\n",
      "API 2.10: https://automat.renci.org/cord19-scigraph/chemical_substance/gene/CHEBI:57947\n",
      "API 2.10 Automat CORD19 Scigraph API: 0 hits\n",
      "API 19.14: https://automat.renci.org/hmdb/chemical_substance/gene/CHEBI:29995\n",
      "API 19.14 Automat HMDB API: 0 hits\n",
      "API 2.11: https://automat.renci.org/cord19-scigraph/chemical_substance/gene/CHEBI:29995\n",
      "API 2.11 Automat CORD19 Scigraph API: 11 hits\n",
      "API 2.12: https://automat.renci.org/cord19-scigraph/chemical_substance/gene/CHEBI:16919\n",
      "API 2.12 Automat CORD19 Scigraph API: 12 hits\n",
      "API 2.13: https://automat.renci.org/cord19-scigraph/chemical_substance/gene/CHEBI:29194\n",
      "API 2.13 Automat CORD19 Scigraph API: 25 hits\n",
      "API 7.10: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:10045/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.10 BioLink API: 14 hits\n",
      "API 2.14: https://automat.renci.org/cord19-scigraph/chemical_substance/gene/CHEBI:24261\n",
      "API 2.14 Automat CORD19 Scigraph API: 42 hits\n",
      "API 14.11 SEMMED Chemical API: 0 hits\n",
      "API 14.12: https://biothings.ncats.io/semmedchemical/query?fields=produced_by (POST -d q=C0020336,C0254211,C0046462,C0030054,C0008269,C0700447,C0003751,C0010286,C0292818,C0674432&scopes=umls)\n",
      "API 14.12 SEMMED Chemical API: 5 hits\n",
      "API 14.13: https://biothings.ncats.io/semmedchemical/query?fields=positively_regulates (POST -d q=C0020336,C0254211,C0046462,C0030054,C0008269,C0700447,C0003751,C0010286,C0292818,C0674432&scopes=umls)\n",
      "API 14.13 SEMMED Chemical API: 512 hits\n",
      "API 16.10: http://ctdbase.org/tools/batchQuery.go?format=json&inputTerms=C112035&inputType=chem&report=genes_curated\n",
      "API 16.10 CTD API: 0 hits\n",
      "API 12.10: https://biothings.ncats.io/semmedgene/query?fields=related_to (POST -d q=C1415900,C1439284,C1428660,C1422064,C1414944,C1825592,C1420826,C1414141,C1415898,C1823096,C1334124,C1333522,C1413931,C1335240,C1415912,C1539099,C1417048,C1333916,C1337092,C1412099,C1416946,C1420310,C1332690,C0081714,C1540024,C1413244,C1417035,C1425041,C0054871,C1413716,C1415274,C1412186,C1426329,C1335818,C1335817,C1419338,C0035094,C1334122,C1420718,C1366571,C0812246,C1412828,C1332036,C1334112,C1367714,C1412332,C1334085,C0021745,C1332807,C1332714,C1413809,C1334098,C1414504,C1335232,C1336641,C0919550,C1334114,C0879590&scopes=umls)\n",
      "API 7.11: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3458/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.11 BioLink API: 128 hits\n",
      "API 7.12: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:920/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.12 BioLink API: 198 hits\n",
      "API 7.13: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3558/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.13 BioLink API: 200 hits\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API 7.14: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:6347/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 12.10 SEMMED Gene API: 0 hits\n",
      "API 7.14 BioLink API: 130 hits\n",
      "API 13.15: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:132943\" AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 13.15 Text Mining Targeted Association API: 0 hits\n",
      "API 13.16: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:33098\" AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 13.16 Text Mining Targeted Association API: 0 hits\n",
      "API 13.17: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:134730\" AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 13.17 Text Mining Targeted Association API: 0 hits\n",
      "API 13.18: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:5801\" AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 13.18 Text Mining Targeted Association API: 32 hits\n",
      "API 13.19: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:48433\" AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 13.19 Text Mining Targeted Association API: 39 hits\n",
      "API 18.15: https://automat.renci.org/chembio/chemical_substance/gene/CHEBI:48433\n",
      "API 18.15 Automat CHEMBIO API: 0 hits\n",
      "API 18.16: https://automat.renci.org/chembio/chemical_substance/gene/CHEBI:132943\n",
      "API 18.16 Automat CHEMBIO API: 0 hits\n",
      "API 1.15: https://automat.renci.org/cord19-scibite/chemical_substance/gene/CHEBI:48433\n",
      "API 1.15 Automat CORD19 Scibite API: 0 hits\n",
      "API 19.15: https://automat.renci.org/hmdb/chemical_substance/gene/CHEBI:132943\n",
      "API 19.15 Automat HMDB API: 0 hits\n",
      "API 1.16: https://automat.renci.org/cord19-scibite/chemical_substance/gene/CHEBI:132943\n",
      "API 1.16 Automat CORD19 Scibite API: 0 hits\n",
      "API 1.17: https://automat.renci.org/cord19-scibite/chemical_substance/gene/CHEBI:31781\n",
      "API 1.17 Automat CORD19 Scibite API: 5 hits\n",
      "API 19.16: https://automat.renci.org/hmdb/chemical_substance/gene/CHEBI:48433\n",
      "API 19.16 Automat HMDB API: 0 hits\n",
      "API 2.15: https://automat.renci.org/cord19-scigraph/chemical_substance/gene/CHEBI:31781\n",
      "API 2.15 Automat CORD19 Scigraph API: 2 hits\n",
      "API 2.16: https://automat.renci.org/cord19-scigraph/chemical_substance/gene/CHEBI:132943\n",
      "API 2.16 Automat CORD19 Scigraph API: 11 hits\n",
      "API 2.17: https://automat.renci.org/cord19-scigraph/chemical_substance/gene/CHEBI:48433\n",
      "API 2.17 Automat CORD19 Scigraph API: 10 hits\n",
      "API 5.15: https://automat.renci.org/pharos/chemical_substance/gene/CHEBI:48433\n",
      "API 5.15 Automat PHAROS API: 0 hits\n",
      "API 5.16: https://automat.renci.org/pharos/chemical_substance/gene/CHEBI:132943\n",
      "API 5.16 Automat PHAROS API: 0 hits\n",
      "API 5.17: https://automat.renci.org/pharos/chemical_substance/gene/CHEBI:31781\n",
      "API 5.17 Automat PHAROS API: 1 hits\n",
      "API 7.15: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:5972/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.15 BioLink API: 33 hits\n",
      "API 7.16: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3553/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.16: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3439/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.16 BioLink API: 35 hits\n",
      "API 7.17: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:92521/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.17 BioLink API: 155 hits\n",
      "API 7.18: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3574/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.18 BioLink API: 57 hits\n",
      "API 7.19 BioLink API: 71 hits\n",
      "API 13.20: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:15379\" AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 13.20 Text Mining Targeted Association API: 0 hits\n",
      "API 13.21: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:27140\" AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 13.21 Text Mining Targeted Association API: 0 hits\n",
      "API 13.22: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:45409\" AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 13.22 Text Mining Targeted Association API: 1 hits\n",
      "API 13.23: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:3638\" AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 13.23: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:25805\" AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 13.23 Text Mining Targeted Association API: 107 hits\n",
      "API 13.24 Text Mining Targeted Association API: 139 hits\n",
      "API 7.20: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:1636/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.20 BioLink API: 38 hits\n",
      "API 7.21: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:213/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.21: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3627/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.21 BioLink API: 199 hits\n",
      "API 7.22 BioLink API: 197 hits\n",
      "API 7.23: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:340024/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.23 BioLink API: 6 hits\n",
      "API 7.24: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3576/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.24 BioLink API: 199 hits\n",
      "API 13.25: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:29194\" AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 13.25 Text Mining Targeted Association API: 0 hits\n",
      "API 13.26: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:16919\" AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 13.26 Text Mining Targeted Association API: 7 hits\n",
      "API 13.27: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:57947\" AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 13.27 Text Mining Targeted Association API: 0 hits\n",
      "API 13.28: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:29995\" AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 13.28 Text Mining Targeted Association API: 22 hits\n",
      "API 13.29: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:24261\" AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 13.29 Text Mining Targeted Association API: 118 hits\n",
      "API 7.25: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:10044/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.25 BioLink API: 13 hits\n",
      "API 7.26: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:10332/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.26 BioLink API: 15 hits\n",
      "API 7.27: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:6868/interactions?direct=True&rows=200&unselect_evidence=True\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API 7.27 BioLink API: 74 hits\n",
      "API 7.28: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:2147/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.28 BioLink API: 200 hits\n",
      "API 7.29: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:2580/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.29 BioLink API: 194 hits\n",
      "API 13.30: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:31781\" AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 13.30 Text Mining Targeted Association API: 0 hits\n",
      "API 13.31: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:132943\" AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 13.31 Text Mining Targeted Association API: 0 hits\n",
      "API 13.32: https://biothings.ncats.io/text_mining_targeted_association/query?fields=object.NCBIgene,association.evidence.publications&q=subject.CHEBI:\"CHEBI:48433\" AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 13.32 Text Mining Targeted Association API: 25 hits\n",
      "API 7.30: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:339967/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.30 BioLink API: 9 hits\n",
      "API 7.31: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:2152/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.31 BioLink API: 41 hits\n",
      "API 7.32: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:682/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.32 BioLink API: 200 hits\n",
      "API 7.33: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:5499/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.33: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7137/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.33 BioLink API: 62 hits\n",
      "API 7.34 BioLink API: 199 hits\n",
      "API 7.35: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:10473/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.35 BioLink API: 12 hits\n",
      "API 7.36: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:1803/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.36 BioLink API: 35 hits\n",
      "API 7.37: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:1514/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.37 BioLink API: 48 hits\n",
      "API 7.38: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:186/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.38: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3605/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.38 BioLink API: 200 hits\n",
      "API 7.39 BioLink API: 105 hits\n",
      "API 7.40: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:4151/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.40 BioLink API: 19 hits\n",
      "API 7.41: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3456/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.41: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3559/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.41 BioLink API: 84 hits\n",
      "API 7.42: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:10410/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.42 BioLink API: 106 hits\n",
      "API 7.43 BioLink API: 145 hits\n",
      "API 7.44: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:2875/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.44 BioLink API: 46 hits\n",
      "API 7.45: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:925/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.45 BioLink API: 52 hits\n",
      "API 7.46: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7124/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.46: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:22848/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.46 BioLink API: 150 hits\n",
      "API 7.47 BioLink API: 200 hits\n",
      "API 7.48: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3586/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.48 BioLink API: 199 hits\n",
      "API 7.49: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3569/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.49 BioLink API: 200 hits\n",
      "API 7.50: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:1506/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.50 BioLink API: 18 hits\n",
      "API 7.51: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:5045/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.51 BioLink API: 59 hits\n",
      "API 7.52: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:54938/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.52 BioLink API: 82 hits\n",
      "API 7.53: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:185/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.53: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:6348/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.53 BioLink API: 60 hits\n",
      "API 7.54 BioLink API: 200 hits\n",
      "API 7.55: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:28/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.55 BioLink API: 13 hits\n",
      "API 7.56: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:6651/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.56: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:54585/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.56 BioLink API: 39 hits\n",
      "API 7.57 BioLink API: 0 hits\n",
      "API 7.58: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7113/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.58: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:59272/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.58 BioLink API: 0 hits\n",
      "API 7.59 BioLink API: 0 hits\n",
      "API 7.60: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7054/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.60 BioLink API: 0 hits\n",
      "API 7.61: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:1401/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.61 BioLink API: 0 hits\n",
      "API 7.62: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:4141/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.62: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:5447/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.62 BioLink API: 0 hits\n",
      "API 7.63 BioLink API: 0 hits\n",
      "API 7.64: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:920/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.64 BioLink API: 0 hits\n",
      "API 7.65: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:10045/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.65 BioLink API: 0 hits\n",
      "API 7.66: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3458/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.66 BioLink API: 0 hits\n",
      "API 7.67: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:6347/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.67: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:92521/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.67 BioLink API: 0 hits\n",
      "API 7.68 BioLink API: 0 hits\n",
      "API 7.69: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3558/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.69 BioLink API: 0 hits\n",
      "API 7.70: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3439/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.70 BioLink API: 0 hits\n",
      "API 7.71: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:5972/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.71 BioLink API: 0 hits\n",
      "API 7.72: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3553/interactions?direct=True&rows=200&unselect_evidence=True\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API 7.72 BioLink API: 0 hits\n",
      "API 7.73: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3574/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.73 BioLink API: 0 hits\n",
      "API 7.74: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:1636/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.74 BioLink API: 0 hits\n",
      "API 7.75: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:340024/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.75 BioLink API: 0 hits\n",
      "API 7.76: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3627/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.76 BioLink API: 0 hits\n",
      "API 7.77: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3576/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.77 BioLink API: 0 hits\n",
      "API 7.78: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:2147/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.78 BioLink API: 0 hits\n",
      "API 7.79: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:213/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.79 BioLink API: 0 hits\n",
      "API 7.80: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:10044/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.80 BioLink API: 0 hits\n",
      "API 7.81: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:6868/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.81 BioLink API: 0 hits\n",
      "API 7.82: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:2580/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.82 BioLink API: 0 hits\n",
      "API 7.83: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:5499/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.83: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:10332/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.83 BioLink API: 0 hits\n",
      "API 7.84 BioLink API: 0 hits\n",
      "API 7.85: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:339967/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.85 BioLink API: 0 hits\n",
      "API 7.86: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7137/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.86 BioLink API: 0 hits\n",
      "API 7.87: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3605/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.87: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:2152/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.87 BioLink API: 0 hits\n",
      "API 7.88 BioLink API: 0 hits\n",
      "API 7.89: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:682/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.89 BioLink API: 0 hits\n",
      "API 7.90: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:1803/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.90 BioLink API: 0 hits\n",
      "API 7.91: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3559/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.91 BioLink API: 0 hits\n",
      "API 7.92: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:186/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.92: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:10473/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.92 BioLink API: 0 hits\n",
      "API 7.93: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:1514/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.93 BioLink API: 0 hits\n",
      "API 7.94 BioLink API: 0 hits\n",
      "API 7.95: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:4151/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.95 BioLink API: 0 hits\n",
      "API 7.96: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3456/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.96 BioLink API: 0 hits\n",
      "API 7.97: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3569/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.97 BioLink API: 0 hits\n",
      "API 7.98: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:10410/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.98: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:2875/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.98 BioLink API: 0 hits\n",
      "API 7.99 BioLink API: 0 hits\n",
      "API 7.100: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:925/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.100 BioLink API: 0 hits\n",
      "API 7.101: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3586/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.101 BioLink API: 0 hits\n",
      "API 7.102: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:185/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.102: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:22848/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.102 BioLink API: 0 hits\n",
      "API 7.103 BioLink API: 0 hits\n",
      "API 7.104: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7124/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.104 BioLink API: 0 hits\n",
      "API 7.105: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:28/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.105 BioLink API: 0 hits\n",
      "API 7.106: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:5045/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.106 BioLink API: 0 hits\n",
      "API 7.107: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:54938/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.107: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:1506/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.107 BioLink API: 0 hits\n",
      "API 7.108 BioLink API: 0 hits\n",
      "API 7.109: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:6348/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.109 BioLink API: 0 hits\n",
      "API 7.110: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:54585/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 7.110 BioLink API: 0 hits\n",
      "\n",
      "BTE starts to perform id-to-object translation.\n",
      "\n",
      "After id-to-object translation, BTE retrieved 10498 unique output nodes.\n",
      "\n",
      "\n",
      "Number of output edges sent to NGD score annotation is 37014. Number of output edges annotated with NGD score is 9491\n",
      "\n",
      "After applying post-query filter, BTE retrieved 2 unique output nodes.\n",
      "\n",
      "========== Query # 3 ==========\n",
      "\n",
      "==== Step #1: Query Path Planning ====\n",
      "\n",
      "Input Types: Gene\n",
      "Output Types: ['PhenotypicFeature', 'Disease', 'BiologicalProcess']\n",
      "Predicates: None\n",
      "\n",
      "BTE found 14 APIs based on SmartAPI Meta-KG.\n",
      "\n",
      "API 1. Automat CORD19 Scibite API (2 API calls)\n",
      "API 2. mydisease.info API (1 API calls)\n",
      "API 3. Automat CORD19 Scigraph API (2 API calls)\n",
      "API 4. TCGA Mutation Frequency KP API (1 API calls)\n",
      "API 5. CORD Gene API (2 API calls)\n",
      "API 6. Automat PHAROS API (2 API calls)\n",
      "API 7. DISEASES API (1 API calls)\n",
      "API 8. BioLink API (2 API calls)\n",
      "API 9. CTD API (1 API calls)\n",
      "API 10. Molecular Data Provider API (1 API calls)\n",
      "API 11. Genetics Provider API (1 API calls)\n",
      "API 12. MyGene.info API (1 API calls)\n",
      "API 13. SEMMED Gene API (32 API calls)\n",
      "API 14. EBIgene2phenotype API (1 API calls)\n",
      "\n",
      "\n",
      "==== Step #2: Query path execution ====\n",
      "\n",
      "NOTE: API requests are dispatched in parallel, so the list of APIs below is ordered by query time.\n",
      "\n",
      "API 14.1: https://pending.biothings.io/ebigene2phenotype/query?fields=gene2phenotype (POST -d q=11892&scopes=_id)\n",
      "API 14.1 EBIgene2phenotype API: 0 hits\n",
      "API 8.1: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7124/diseases?direct=True&rows=200&unselect_evidence=True\n",
      "API 8.1 BioLink API: 3 hits\n",
      "API 4.1: https://biothings.ncats.io/tcga_mut_freq_kp/query?fields=association.freq_by_sample,object.MONDO&q=subject.SYMBOL:'TNF'%20AND%20association.freq_by_sample:>0.1&size=1000&sort=-association.freq_by_sample\n",
      "API 4.1 TCGA Mutation Frequency KP API: 0 hits\n",
      "API 12.1: https://mygene.info/v3/query?fields=go.BP (POST -d q=7124&scopes=entrezgene)\n",
      "API 12.1 MyGene.info API: 177 hits\n",
      "API 9.1: http://ctdbase.org/tools/batchQuery.go?format=json&inputTerms=7124&inputType=gene&report=diseases_curated\n",
      "API 5.1: https://biothings.ncats.io/cord_gene/query?fields=associated_with (POST -d q=11892&scopes=hgnc)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API 9.1 CTD API: 178 hits\n",
      "API 5.1: https://biothings.ncats.io/cord_gene/query?fields=associated_with (POST -d q=11892&scopes=hgnc)\n",
      "API 10.1: https://translator.broadinstitute.org/molepro_reasoner/query (POST -d {'message': {'query_graph': {'edges': [{'id': 'e00', 'source_id': 'n00', 'target_id': 'n01', 'type': 'related_to'}], 'nodes': [{'curie': 'HGNC:11892', 'id': 'n00', 'type': 'gene'}, {'id': 'n01', 'type': 'disease'}]}}})\n",
      "API 10.1 Molecular Data Provider API: 0 hits\n",
      "API 11.1: https://translator.broadinstitute.org/genetics_data_provider/query (POST -d {'message': {'query_graph': {'edges': [{'id': 'e00', 'source_id': 'n00', 'target_id': 'n01', 'type': 'associated'}], 'nodes': [{'curie': 'NCBIGene:7124', 'id': 'n00', 'type': 'gene'}, {'id': 'n01', 'type': 'disease'}]}}})\n",
      "API 8.2: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7124/phenotypes?direct=True&rows=200&unselect_evidence=True\n",
      "API 8.2 BioLink API: 0 hits\n",
      "API 7.1: https://pending.biothings.io/DISEASES/query?fields=DISEASES.doid&size=250 (POST -d q=TNF&scopes=DISEASES.associatedWith.symbol)\n",
      "API 1.1: https://automat.renci.org/cord19-scibite/gene/disease/NCBIGene:7124\n",
      "API 1.1: https://automat.renci.org/cord19-scibite/gene/disease/NCBIGene:7124\n",
      "API 2.1: http://mydisease.info/v1/query?fields=disgenet.xrefs.umls&size=250 (POST -d q=7124&scopes=disgenet.genes_related_to_disease.gene_id)\n",
      "API 2.1 mydisease.info API: 212 hits\n",
      "API 1.1 Automat CORD19 Scibite API: 44 hits\n",
      "API 1.2 Automat CORD19 Scibite API: 0 hits\n",
      "API 3.1: https://automat.renci.org/cord19-scigraph/gene/disease/NCBIGene:7124\n",
      "API 3.1 Automat CORD19 Scigraph API: 0 hits\n",
      "API 3.2: https://automat.renci.org/cord19-scigraph/gene/disease/NCBIGene:7124\n",
      "API 3.2 Automat CORD19 Scigraph API: 153 hits\n",
      "API 11.1 Genetics Provider API: 32 hits\n",
      "API 7.1 DISEASES API: 250 hits\n",
      "API 13.1: https://biothings.ncats.io/semmedgene/query?fields=affected_by (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.1 SEMMED Gene API: 579 hits\n",
      "API 13.2: https://biothings.ncats.io/semmedgene/query?fields=affects (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.2: https://biothings.ncats.io/semmedgene/query?fields=affected_by (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 6.1: https://automat.renci.org/pharos/gene/disease/NCBIGene:7124\n",
      "API 6.1 Automat PHAROS API: 3 hits\n",
      "API 6.2: https://automat.renci.org/pharos/gene/disease/NCBIGene:7124\n",
      "API 6.2 Automat PHAROS API: 0 hits\n",
      "API 13.2: https://biothings.ncats.io/semmedgene/query?fields=affects (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API call to CORD Gene API with input ['11892'] failed with unknown response\n",
      "API call to CORD Gene API with input ['11892'] failed with unknown response\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 297, in callSingleAPI\n",
      "    result = tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/__init__.py\", line 49, in transform\n",
      "    return self.tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/base_transformer.py\", line 78, in transform\n",
      "    item = self.wrap(item)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/cord_transformer.py\", line 16, in wrap\n",
      "    for k in item.keys():\n",
      "RuntimeError: dictionary keys changed during iteration\n",
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 297, in callSingleAPI\n",
      "    result = tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/__init__.py\", line 49, in transform\n",
      "    return self.tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/base_transformer.py\", line 78, in transform\n",
      "    item = self.wrap(item)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/cord_transformer.py\", line 16, in wrap\n",
      "    for k in item.keys():\n",
      "RuntimeError: dictionary keys changed during iteration\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API 13.2 SEMMED Gene API: 1681 hits\n",
      "API 13.3 SEMMED Gene API: 6 hits\n",
      "API 13.4 SEMMED Gene API: 2307 hits\n",
      "API 13.5: https://biothings.ncats.io/semmedgene/query?fields=disrupted_by (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.5 SEMMED Gene API: 0 hits\n",
      "API 13.6: https://biothings.ncats.io/semmedgene/query?fields=disrupted_by (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.6 SEMMED Gene API: 0 hits\n",
      "API 13.7: https://biothings.ncats.io/semmedgene/query?fields=causes (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.7: https://biothings.ncats.io/semmedgene/query?fields=causes (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.7: https://biothings.ncats.io/semmedgene/query?fields=affects (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.7 SEMMED Gene API: 360 hits\n",
      "API 13.8 SEMMED Gene API: 2390 hits\n",
      "API 13.9 SEMMED Gene API: 1 hits\n",
      "API 13.10: https://biothings.ncats.io/semmedgene/query?fields=disrupts (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.10: https://biothings.ncats.io/semmedgene/query?fields=disrupts (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.10: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulated_by (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.10: https://biothings.ncats.io/semmedgene/query?fields=disrupts (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.10: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulated_by (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.10 SEMMED Gene API: 562 hits\n",
      "API 13.11 SEMMED Gene API: 2 hits\n",
      "API 13.12 SEMMED Gene API: 666 hits\n",
      "API 13.13 SEMMED Gene API: 1 hits\n",
      "API 13.14 SEMMED Gene API: 1 hits\n",
      "API 13.15: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulates (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.15: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulates (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.15: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulates (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.15: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulated_by (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.15 SEMMED Gene API: 1 hits\n",
      "API 13.16: https://biothings.ncats.io/semmedgene/query?fields=physically_interacts_with (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.16 SEMMED Gene API: 0 hits\n",
      "API 13.17 SEMMED Gene API: 0 hits\n",
      "API 13.18 SEMMED Gene API: 1 hits\n",
      "API 13.19 SEMMED Gene API: 1 hits\n",
      "API 13.20: https://biothings.ncats.io/semmedgene/query?fields=positively_regulated_by (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.20: https://biothings.ncats.io/semmedgene/query?fields=positively_regulated_by (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.20: https://biothings.ncats.io/semmedgene/query?fields=positively_regulated_by (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.20: https://biothings.ncats.io/semmedgene/query?fields=physically_interacts_with (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.20: https://biothings.ncats.io/semmedgene/query?fields=physically_interacts_with (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.20 SEMMED Gene API: 1 hits\n",
      "API 13.21 SEMMED Gene API: 0 hits\n",
      "API 13.22 SEMMED Gene API: 2 hits\n",
      "API 13.23 SEMMED Gene API: 2 hits\n",
      "API 13.24 SEMMED Gene API: 1 hits\n",
      "API 13.25: https://biothings.ncats.io/semmedgene/query?fields=prevents (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.25: https://biothings.ncats.io/semmedgene/query?fields=positively_regulates (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.25 SEMMED Gene API: 215 hits\n",
      "API 13.26: https://biothings.ncats.io/semmedgene/query?fields=positively_regulates (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.26: https://biothings.ncats.io/semmedgene/query?fields=related_to (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.26: https://biothings.ncats.io/semmedgene/query?fields=positively_regulates (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.26 SEMMED Gene API: 1 hits\n",
      "API 13.27 SEMMED Gene API: 0 hits\n",
      "API 13.28 SEMMED Gene API: 8 hits\n",
      "API 13.29 SEMMED Gene API: 7 hits\n",
      "API 13.30: https://biothings.ncats.io/semmedgene/query?fields=treats (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.30: https://biothings.ncats.io/semmedgene/query?fields=related_to (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.30: https://biothings.ncats.io/semmedgene/query?fields=related_to (POST -d q=C0017337,C0812246&scopes=umls)\n",
      "API 13.30 SEMMED Gene API: 1933 hits\n",
      "API 13.31 SEMMED Gene API: 0 hits\n",
      "API 13.32 SEMMED Gene API: 3741 hits\n",
      "\n",
      "BTE starts to perform id-to-object translation.\n",
      "\n",
      "After id-to-object translation, BTE retrieved 7203 unique output nodes.\n",
      "\n",
      "\n",
      "Number of output edges sent to NGD score annotation is 15522. Number of output edges annotated with NGD score is 1832\n",
      "\n",
      "==========\n",
      "========== Final assembly of results ==========\n",
      "==========\n",
      "\n",
      "In the #1 query, BTE found 69 unique nodes.\n",
      "In the #2 query, BTE found 2 unique nodes.\n",
      "In the #3 query, BTE found 7203 unique nodes.\n"
     ]
    }
   ],
   "source": [
    "## way to filter at the gene step for top x number of genes? \n",
    "pd2 = Predict(\n",
    "    input_objs=[disease],\n",
    "    intermediate_nodes =[['Gene', 'SequenceVariant', 'ChemicalSubstance', 'Disease', \n",
    "                'MolecularActivity', 'BiologicalProcess', 'CellularComponent', \n",
    "                'Pathway', 'AnatomicalEntity', 'PhenotypicFeature'],'Gene'], \n",
    "    output_types =['PhenotypicFeature', 'Disease', 'BiologicalProcess'], \n",
    "    config= query_config \n",
    ")\n",
    "pd2.connect(verbose=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1905047, 27)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2 = pd2.display_table_view()\n",
    "df2.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(806, 27)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# filter gene -> symptoms table to only include rows where symptoms match disease symptoms\n",
    "indices_with_symptom_outputs = [i for i, val in enumerate([x.upper() for x in list(df2['output_label'])]) if val in disease_symptoms]\n",
    "indirect_relevant_genes_to_symptoms_df = df2.iloc[indices_with_symptom_outputs]\n",
    "indirect_relevant_genes_to_symptoms_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'dioxygen': 60,\n",
       " 'chloroquine': 54,\n",
       " 'REN': 52,\n",
       " 'IL2': 34,\n",
       " 'IFNA1': 34,\n",
       " 'IL6': 34,\n",
       " 'CCL2': 32,\n",
       " 'CD4': 30,\n",
       " 'TNF': 30,\n",
       " 'IL10': 30,\n",
       " 'ALB': 28,\n",
       " 'IL2RA': 28,\n",
       " 'CCL3': 24,\n",
       " 'hydroxychloroquine': 24,\n",
       " 'ACE': 20,\n",
       " 'CXCL8': 20,\n",
       " 'TH': 18,\n",
       " 'IL1B': 18,\n",
       " 'IFNB1': 18,\n",
       " 'AGTR1': 16}"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indirect_genes = \tlist(indirect_relevant_genes_to_symptoms_df[\"node1_label\"])\n",
    "indirect_gene_dict = {x:indirect_genes.count(x) for x in indirect_genes}\n",
    "indirect_gene_dict = dict(sorted(indirect_gene_dict.items(), key=lambda x: x[1], reverse=True)[:max_direct])\n",
    "indirect_gene_dict"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Top Genes -> Pathways "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get gene inputs through hint module\n",
    "gene_inputs = []\n",
    "for gene in list(direct_gene_dict.keys()): \n",
    "    try: \n",
    "        gene_input = ht.query(gene)[\"Gene\"][0]\n",
    "        gene_inputs.append(gene_input)\n",
    "    except: \n",
    "        print(gene + ' Failed')\n",
    "\n",
    "# print(gene_inputs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==========\n",
      "========== QUERY PARAMETER SUMMARY ==========\n",
      "==========\n",
      "\n",
      "Your query have 6 input nodes, including tumor necrosis factor,interleukin 2,interleukin 1 beta,interferon alpha 1,troponin I3, cardiac type .... And BTE will find paths that connect your input nodes to your output types ['Pathway']. Paths will contain 1 intermediate nodes.\n",
      "\n",
      "Intermediate node #1 will have these type constraints: None\n",
      "\n",
      "========== Query # 1 ==========\n",
      "\n",
      "==== Step #1: Query Path Planning ====\n",
      "\n",
      "Input Types: Gene\n",
      "Output Types: None\n",
      "Predicates: None\n",
      "\n",
      "BTE found 21 APIs based on SmartAPI Meta-KG.\n",
      "\n",
      "API 1. Text Mining Targeted Association API (12 API calls)\n",
      "API 2. Automat CORD19 Scibite API (24 API calls)\n",
      "API 3. mydisease.info API (1 API calls)\n",
      "API 4. Automat CORD19 Scigraph API (24 API calls)\n",
      "API 5. TCGA Mutation Frequency KP API (1 API calls)\n",
      "API 6. CORD Gene API (11 API calls)\n",
      "API 7. Automat PHAROS API (24 API calls)\n",
      "API 8. MyChem.info API (3 API calls)\n",
      "API 9. DISEASES API (1 API calls)\n",
      "API 10. BioLink API (30 API calls)\n",
      "API 11. OpenTarget API (6 API calls)\n",
      "API 12. CTD API (6 API calls)\n",
      "API 13. Molecular Data Provider API (6 API calls)\n",
      "API 14. Genetics Provider API (6 API calls)\n",
      "API 15. MyGene.info API (9 API calls)\n",
      "API 16. Automat CHEMBIO API (12 API calls)\n",
      "API 17. Automat HMDB API (12 API calls)\n",
      "API 18. SEMMED Gene API (74 API calls)\n",
      "API 19. EBIgene2phenotype API (1 API calls)\n",
      "API 20. Drug Response KP API (6 API calls)\n",
      "API 21. BioThings DGIdb API (1 API calls)\n",
      "\n",
      "\n",
      "==== Step #2: Query path execution ====\n",
      "\n",
      "NOTE: API requests are dispatched in parallel, so the list of APIs below is ordered by query time.\n",
      "\n",
      "API 15.1: https://mygene.info/v3/query?fields=go.CC (POST -d q=7124,3553,3439,3569,7137,3558&scopes=entrezgene)\n",
      "API 15.1 MyGene.info API: 34 hits\n",
      "API 15.2: https://mygene.info/v3/query?fields=go.BP (POST -d q=7124,3553,3439,3569,7137,3558&scopes=entrezgene)\n",
      "API 15.2 MyGene.info API: 463 hits\n",
      "API 1.1: https://biothings.ncats.io/text_mining_targeted_association/query?fields=subject.CHEBI,association.evidence.publications&q=object.NCBIgene:3439 AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 1.1 Text Mining Targeted Association API: 0 hits\n",
      "API 11.1: https://platform-api.opentargets.io/v3/platform/public/evidence/filter?datasource=chembl&fields=drug&size=100&target=ENSG00000197919\n",
      "API 11.1 OpenTarget API: 10 hits\n",
      "API 11.2: https://platform-api.opentargets.io/v3/platform/public/evidence/filter?datasource=chembl&fields=drug&size=100&target=ENSG00000125538\n",
      "API 11.2 OpenTarget API: 100 hits\n",
      "API 8.1: https://mychem.info/v1/query?fields=chembl.molecule_chembl_id&size=250 (POST -d q=TNF,IFNA1,IL2,IL6,TNNI3,IL1B&scopes=drugcentral.bioactivity.uniprot.gene_symbol)\n",
      "API 8.1 MyChem.info API: 5 hits\n",
      "API 8.2: https://mychem.info/v1/query?fields=drugbank.id&size=250 (POST -d q=TNF,IFNA1,IL2,IL6,TNNI3,IL1B&scopes=drugbank.enzymes.gene_name)\n",
      "API 8.2 MyChem.info API: 0 hits\n",
      "API 1.2: https://biothings.ncats.io/text_mining_targeted_association/query?fields=subject.CHEBI,association.evidence.publications&q=object.NCBIgene:7124 AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 15.3: https://mygene.info/v3/query?fields=pathway.wikipathways (POST -d q=7124,3553,3439,3569,7137,3558&scopes=entrezgene)\n",
      "API 15.3 MyGene.info API: 156 hits\n",
      "API 20.1: https://biothings.ncats.io/drug_response_kp/query?fields=association.context.disease.mondo,object.PUBCHEM,association.effect_size,association.pvalue&q=subject.NCBIGene:3553%20AND%20association.effect_size:<0%20AND%20association.pvalue:<0.05&size=1000&sort=association.pvalue\n",
      "API 20.1 Drug Response KP API: 0 hits\n",
      "API 5.1: https://biothings.ncats.io/tcga_mut_freq_kp/query?fields=association.freq_by_sample,object.MONDO&q=subject.SYMBOL:'TNF,IFNA1,IL2,IL6,TNNI3,IL1B'%20AND%20association.freq_by_sample:>0.1&size=1000&sort=-association.freq_by_sample\n",
      "API 5.1 TCGA Mutation Frequency KP API: 0 hits\n",
      "API 20.2: https://biothings.ncats.io/drug_response_kp/query?fields=association.context.disease.mondo,object.PUBCHEM,association.effect_size,association.pvalue&q=subject.NCBIGene:7124%20AND%20association.effect_size:<0%20AND%20association.pvalue:<0.05&size=1000&sort=association.pvalue\n",
      "API 20.2 Drug Response KP API: 0 hits\n",
      "API 20.3: https://biothings.ncats.io/drug_response_kp/query?fields=association.context.disease.mondo,object.PUBCHEM,association.effect_size,association.pvalue&q=subject.NCBIGene:3439%20AND%20association.effect_size:<0%20AND%20association.pvalue:<0.05&size=1000&sort=association.pvalue\n",
      "API 20.3 Drug Response KP API: 0 hits\n",
      "API 1.2: https://biothings.ncats.io/text_mining_targeted_association/query?fields=subject.CHEBI,association.evidence.publications&q=object.NCBIgene:3558 AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 1.2 Text Mining Targeted Association API: 53 hits\n",
      "API 12.1: http://ctdbase.org/tools/batchQuery.go?format=json&inputTerms=3553&inputType=gene&report=diseases_curated\n",
      "API 12.1 CTD API: 99 hits\n",
      "API 1.3: https://biothings.ncats.io/text_mining_targeted_association/query?fields=subject.CHEBI,association.evidence.publications&q=object.NCBIgene:3553 AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 1.3 Text Mining Targeted Association API: 25 hits\n",
      "API 18.1: https://biothings.ncats.io/semmedgene/query?fields=affected_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 20.4: https://biothings.ncats.io/drug_response_kp/query?fields=association.context.disease.mondo,object.PUBCHEM,association.effect_size,association.pvalue&q=subject.NCBIGene:3558%20AND%20association.effect_size:<0%20AND%20association.pvalue:<0.05&size=1000&sort=association.pvalue\n",
      "API 20.4 Drug Response KP API: 0 hits\n",
      "API 18.1: https://biothings.ncats.io/semmedgene/query?fields=affected_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 12.2: http://ctdbase.org/tools/batchQuery.go?format=json&inputTerms=3558&inputType=gene&report=diseases_curated\n",
      "API 12.2 CTD API: 36 hits\n",
      "API 8.3: https://mychem.info/v1/query?fields=drugbank.id&size=250 (POST -d q=TNF,IFNA1,IL2,IL6,TNNI3,IL1B&scopes=drugbank.targets.gene_name)\n",
      "API 8.3 MyChem.info API: 80 hits\n",
      "API 12.3: http://ctdbase.org/tools/batchQuery.go?format=json&inputTerms=3439&inputType=gene&report=diseases_curated\n",
      "API 12.3 CTD API: 14 hits\n",
      "API 11.3: https://platform-api.opentargets.io/v3/platform/public/evidence/filter?datasource=chembl&fields=drug&size=100&target=ENSG00000204490\n",
      "API 11.3 OpenTarget API: 0 hits\n",
      "API 11.4: https://platform-api.opentargets.io/v3/platform/public/evidence/filter?datasource=chembl&fields=drug&size=100&target=ENSG00000109471\n",
      "API 11.4 OpenTarget API: 0 hits\n",
      "API 1.4 Text Mining Targeted Association API: 234 hits\n",
      "API 2.1: https://automat.renci.org/cord19-scibite/gene/chemical_substance/NCBIGene:7124\n",
      "API 2.1 Automat CORD19 Scibite API: 37 hits\n",
      "API 2.2: https://automat.renci.org/cord19-scibite/gene/chemical_substance/NCBIGene:3558\n",
      "API 2.2 Automat CORD19 Scibite API: 8 hits\n",
      "API 7.1: https://automat.renci.org/pharos/gene/chemical_substance/NCBIGene:7124\n",
      "API 7.1 Automat PHAROS API: 300 hits\n",
      "API 2.3: https://automat.renci.org/cord19-scibite/gene/chemical_substance/NCBIGene:3439\n",
      "API 2.3 Automat CORD19 Scibite API: 14 hits\n",
      "API 4.1: https://automat.renci.org/cord19-scigraph/gene/chemical_substance/NCBIGene:3439\n",
      "API 4.1 Automat CORD19 Scigraph API: 0 hits\n",
      "API 16.1: https://automat.renci.org/chembio/gene/chemical_substance/NCBIGene:3553\n",
      "API 16.1 Automat CHEMBIO API: 0 hits\n",
      "API 4.2: https://automat.renci.org/cord19-scigraph/gene/chemical_substance/NCBIGene:7124\n",
      "API 4.2 Automat CORD19 Scigraph API: 0 hits\n",
      "API 7.2: https://automat.renci.org/pharos/gene/chemical_substance/NCBIGene:3553\n",
      "API 7.2 Automat PHAROS API: 0 hits\n",
      "API 16.2: https://automat.renci.org/chembio/gene/chemical_substance/NCBIGene:3558\n",
      "API 16.2 Automat CHEMBIO API: 0 hits\n",
      "API 17.1: https://automat.renci.org/hmdb/gene/chemical_substance/NCBIGene:3558\n",
      "API 17.1 Automat HMDB API: 0 hits\n",
      "API 2.4: https://automat.renci.org/cord19-scibite/gene/chemical_substance/NCBIGene:3553\n",
      "API 2.4 Automat CORD19 Scibite API: 20 hits\n",
      "API 7.3: https://automat.renci.org/pharos/gene/chemical_substance/NCBIGene:3558\n",
      "API 7.3 Automat PHAROS API: 11 hits\n",
      "API 16.3: https://automat.renci.org/chembio/gene/chemical_substance/NCBIGene:7124\n",
      "API 16.3 Automat CHEMBIO API: 0 hits\n",
      "API 17.2: https://automat.renci.org/hmdb/gene/chemical_substance/NCBIGene:3439\n",
      "API 17.2 Automat HMDB API: 0 hits\n",
      "API 16.4: https://automat.renci.org/chembio/gene/chemical_substance/NCBIGene:3439\n",
      "API 16.4 Automat CHEMBIO API: 0 hits\n",
      "API 4.3: https://automat.renci.org/cord19-scigraph/gene/chemical_substance/NCBIGene:3558\n",
      "API 4.3 Automat CORD19 Scigraph API: 0 hits\n",
      "API 4.4: https://automat.renci.org/cord19-scigraph/gene/chemical_substance/NCBIGene:3553\n",
      "API 4.4 Automat CORD19 Scigraph API: 0 hits\n",
      "API 17.3: https://automat.renci.org/hmdb/gene/chemical_substance/NCBIGene:3553\n",
      "API 17.3 Automat HMDB API: 0 hits\n",
      "API 17.4: https://automat.renci.org/hmdb/gene/chemical_substance/NCBIGene:7124\n",
      "API 17.4 Automat HMDB API: 0 hits\n",
      "API 7.4: https://automat.renci.org/pharos/gene/chemical_substance/NCBIGene:3439\n",
      "API 7.4 Automat PHAROS API: 0 hits\n",
      "API 18.1 SEMMED Gene API: 275 hits\n",
      "API 18.2 SEMMED Gene API: 3 hits\n",
      "API 10.1: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7124/anatomy?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.1 BioLink API: 20 hits\n",
      "API 10.2: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3553/anatomy?direct=True&rows=200&unselect_evidence=True\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API 10.2 BioLink API: 20 hits\n",
      "API 3.1: http://mydisease.info/v1/query?fields=disgenet.xrefs.umls&size=250 (POST -d q=7124,3553,3439,3569,7137,3558&scopes=disgenet.genes_related_to_disease.gene_id)\n",
      "API 3.1 mydisease.info API: 582 hits\n",
      "API 10.3: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3439/anatomy?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.3 BioLink API: 19 hits\n",
      "API 10.4: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3558/anatomy?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.4 BioLink API: 20 hits\n",
      "API 6.1: https://biothings.ncats.io/cord_gene/query?fields=associated_with (POST -d q=11892,5992,6001,5417,11947,6018&scopes=hgnc)\n",
      "API 19.1: https://pending.biothings.io/ebigene2phenotype/query?fields=gene2phenotype (POST -d q=11892,5992,6001,5417,11947,6018&scopes=_id)\n",
      "API 19.1 EBIgene2phenotype API: 0 hits\n",
      "API 12.4: http://ctdbase.org/tools/batchQuery.go?format=json&inputTerms=7124&inputType=gene&report=diseases_curated\n",
      "API 12.4 CTD API: 178 hits\n",
      "API 6.1: https://biothings.ncats.io/cord_gene/query?fields=associated_with (POST -d q=11892,5992,6001,5417,11947,6018&scopes=hgnc)\n",
      "API 6.1: https://biothings.ncats.io/cord_gene/query?fields=associated_with (POST -d q=11892,5992,6001,5417,11947,6018&scopes=hgnc)\n",
      "API 6.1: https://biothings.ncats.io/cord_gene/query?fields=associated_with (POST -d q=11892,5992,6001,5417,11947,6018&scopes=hgnc)\n",
      "API 18.3: https://biothings.ncats.io/semmedgene/query?fields=affected_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.3 SEMMED Gene API: 5 hits\n",
      "API 21.1: https://biothings.ncats.io/dgidb/query?fields=object.CHEMBL_COMPOUND,association.provided_by,association.pubmed&size=1000 (POST -d q=7124,3553,3439,3569,7137,3558&scopes=subject.NCBIGene)\n",
      "API 21.1 BioThings DGIdb API: 188 hits\n",
      "API 18.4: https://biothings.ncats.io/semmedgene/query?fields=affected_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.4 SEMMED Gene API: 7 hits\n",
      "API 9.1: https://pending.biothings.io/DISEASES/query?fields=DISEASES.doid&size=250 (POST -d q=TNF,IFNA1,IL2,IL6,TNNI3,IL1B&scopes=DISEASES.associatedWith.symbol)\n",
      "API 15.4: https://mygene.info/v3/query?fields=go.MF (POST -d q=7124,3553,3439,3569,7137,3558&scopes=entrezgene)\n",
      "API 15.4 MyGene.info API: 46 hits\n",
      "API call to Molecular Data Provider API with input 5417 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Molecular Data Provider API with input 5992 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Molecular Data Provider API with input 6001 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Molecular Data Provider API with input 11892 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Genetics Provider API with input 3439 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Genetics Provider API with input 7124 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Genetics Provider API with input 3553 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API call to Genetics Provider API with input 3558 failed with timeout error. Current timeout limit is 5 seconds\n",
      "API 9.1 DISEASES API: 1166 hits\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 297, in callSingleAPI\n",
      "    result = tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/__init__.py\", line 49, in transform\n",
      "    return self.tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/base_transformer.py\", line 78, in transform\n",
      "    item = self.wrap(item)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/cord_transformer.py\", line 16, in wrap\n",
      "    for k in item.keys():\n",
      "RuntimeError: dictionary keys changed during iteration\n",
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 297, in callSingleAPI\n",
      "    result = tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/__init__.py\", line 49, in transform\n",
      "    return self.tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/base_transformer.py\", line 78, in transform\n",
      "    item = self.wrap(item)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/cord_transformer.py\", line 16, in wrap\n",
      "    for k in item.keys():\n",
      "RuntimeError: dictionary keys changed during iteration\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API call to CORD Gene API with input ['11892', '5992', '6001', '5417', '11947', '6018'] failed with unknown response\n",
      "API call to CORD Gene API with input ['11892', '5992', '6001', '5417', '11947', '6018'] failed with unknown response\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 297, in callSingleAPI\n",
      "    result = tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/__init__.py\", line 49, in transform\n",
      "    return self.tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/base_transformer.py\", line 78, in transform\n",
      "    item = self.wrap(item)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/cord_transformer.py\", line 16, in wrap\n",
      "    for k in item.keys():\n",
      "RuntimeError: dictionary keys changed during iteration\n",
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 297, in callSingleAPI\n",
      "    result = tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/__init__.py\", line 49, in transform\n",
      "    return self.tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/base_transformer.py\", line 78, in transform\n",
      "    item = self.wrap(item)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/cord_transformer.py\", line 16, in wrap\n",
      "    for k in item.keys():\n",
      "RuntimeError: dictionary keys changed during iteration\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API call to CORD Gene API with input ['11892', '5992', '6001', '5417', '11947', '6018'] failed with unknown response\n",
      "API call to CORD Gene API with input ['11892', '5992', '6001', '5417', '11947', '6018'] failed with unknown response\n",
      "API 1.5: https://biothings.ncats.io/text_mining_targeted_association/query?fields=subject.CHEBI,association.evidence.publications&q=object.NCBIgene:7137 AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 1.5 Text Mining Targeted Association API: 5 hits\n",
      "API 1.6: https://biothings.ncats.io/text_mining_targeted_association/query?fields=subject.CHEBI,association.evidence.publications&q=object.NCBIgene:3558 AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 1.6: https://biothings.ncats.io/text_mining_targeted_association/query?fields=subject.CHEBI,association.evidence.publications&q=object.NCBIgene:3553 AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 1.6 Text Mining Targeted Association API: 34 hits\n",
      "API 15.5: https://mygene.info/v3/query?fields=pathway.reactome&species=human (POST -d q=7124,3553,3439,3569,7137,3558&scopes=entrezgene)\n",
      "API 15.5 MyGene.info API: 100 hits\n",
      "API 1.7: https://biothings.ncats.io/text_mining_targeted_association/query?fields=subject.CHEBI,association.evidence.publications&q=object.NCBIgene:3569 AND association.edge_label:positively_regulates_entity_to_entity&size=1000\n",
      "API 11.5: https://platform-api.opentargets.io/v3/platform/public/evidence/filter?datasource=chembl&fields=drug&size=100&target=ENSG00000136244\n",
      "API 11.5 OpenTarget API: 70 hits\n",
      "API 11.6: https://platform-api.opentargets.io/v3/platform/public/evidence/filter?datasource=chembl&fields=drug&size=100&target=ENSG00000129991\n",
      "API 11.6 OpenTarget API: 44 hits\n",
      "API 1.7 Text Mining Targeted Association API: 103 hits\n",
      "API 10.5: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3569/anatomy?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.5 BioLink API: 19 hits\n",
      "API 1.8: https://biothings.ncats.io/text_mining_targeted_association/query?fields=subject.CHEBI,association.evidence.publications&q=object.NCBIgene:7124 AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 10.6: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7137/anatomy?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.6 BioLink API: 20 hits\n",
      "API 1.8 Text Mining Targeted Association API: 256 hits\n",
      "API 1.9 Text Mining Targeted Association API: 421 hits\n",
      "API 12.5: http://ctdbase.org/tools/batchQuery.go?format=json&inputTerms=7137&inputType=gene&report=diseases_curated\n",
      "API 12.5 CTD API: 11 hits\n",
      "API 12.6: http://ctdbase.org/tools/batchQuery.go?format=json&inputTerms=3569&inputType=gene&report=diseases_curated\n",
      "API 12.6 CTD API: 123 hits\n",
      "API 6.1: https://biothings.ncats.io/cord_gene/query?fields=associated_with (POST -d q=11892,5992,6001,5417,11947,6018&scopes=hgnc)\n",
      "API 6.1: https://biothings.ncats.io/cord_gene/query?fields=associated_with (POST -d q=11892,5992,6001,5417,11947,6018&scopes=hgnc)\n",
      "API 20.5: https://biothings.ncats.io/drug_response_kp/query?fields=association.context.disease.mondo,object.PUBCHEM,association.effect_size,association.pvalue&q=subject.NCBIGene:3569%20AND%20association.effect_size:<0%20AND%20association.pvalue:<0.05&size=1000&sort=association.pvalue\n",
      "API 20.5 Drug Response KP API: 0 hits\n",
      "API 20.6: https://biothings.ncats.io/drug_response_kp/query?fields=association.context.disease.mondo,object.PUBCHEM,association.effect_size,association.pvalue&q=subject.NCBIGene:7137%20AND%20association.effect_size:<0%20AND%20association.pvalue:<0.05&size=1000&sort=association.pvalue\n",
      "API 20.6 Drug Response KP API: 0 hits\n",
      "API 15.6: https://mygene.info/v3/query?fields=uniprot.Swiss-Prot (POST -d q=ENSG00000204490,ENSG00000109471,ENSG00000136244,ENSG00000129991,ENSG00000197919,ENSG00000125538&scopes=ensembl.gene)\n",
      "API 15.6 MyGene.info API: 6 hits\n",
      "API 15.7: https://mygene.info/v3/query?fields=ensembl.transcript (POST -d q=ENSG00000204490,ENSG00000109471,ENSG00000136244,ENSG00000129991,ENSG00000197919,ENSG00000125538&scopes=ensembl.gene)\n",
      "API 15.7 MyGene.info API: 38 hits\n",
      "API 15.8: https://mygene.info/v3/query?fields=pantherdb.ortholog (POST -d q=7124,3553,3439,3569,7137,3558&scopes=entrezgene)\n",
      "API 15.8 MyGene.info API: 18 hits\n",
      "API 15.9: https://mygene.info/v3/query?fields=ensembl.protein (POST -d q=ENSG00000204490,ENSG00000109471,ENSG00000136244,ENSG00000129991,ENSG00000197919,ENSG00000125538&scopes=ensembl.gene)\n",
      "API 15.9 MyGene.info API: 25 hits\n",
      "API 18.5: https://biothings.ncats.io/semmedgene/query?fields=affects (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 10.7: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7124/diseases?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.7 BioLink API: 3 hits\n",
      "API 10.8: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3553/diseases?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.8 BioLink API: 5 hits\n",
      "API 10.9: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3558/diseases?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.9 BioLink API: 4 hits\n",
      "API 6.1: https://biothings.ncats.io/cord_gene/query?fields=associated_with (POST -d q=11892,5992,6001,5417,11947,6018&scopes=hgnc)\n",
      "API 6.1: https://biothings.ncats.io/cord_gene/query?fields=associated_with (POST -d q=11892,5992,6001,5417,11947,6018&scopes=hgnc)\n",
      "API 18.5: https://biothings.ncats.io/semmedgene/query?fields=affects (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 6.1: https://biothings.ncats.io/cord_gene/query?fields=associated_with (POST -d q=11892,5992,6001,5417,11947,6018&scopes=hgnc)\n",
      "API 18.5: https://biothings.ncats.io/semmedgene/query?fields=affected_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.5: https://biothings.ncats.io/semmedgene/query?fields=affects (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 7.5: https://automat.renci.org/pharos/gene/chemical_substance/NCBIGene:3553\n",
      "API 7.5 Automat PHAROS API: 0 hits\n",
      "API 4.5: https://automat.renci.org/cord19-scigraph/gene/chemical_substance/NCBIGene:3558\n",
      "API 4.5 Automat CORD19 Scigraph API: 7 hits\n",
      "API 4.6: https://automat.renci.org/cord19-scigraph/gene/chemical_substance/NCBIGene:3553\n",
      "API 4.6 Automat CORD19 Scigraph API: 11 hits\n",
      "API 16.5: https://automat.renci.org/chembio/gene/chemical_substance/NCBIGene:3558\n",
      "API 16.5 Automat CHEMBIO API: 26 hits\n",
      "API 7.6: https://automat.renci.org/pharos/gene/chemical_substance/NCBIGene:7137\n",
      "API 7.6 Automat PHAROS API: 0 hits\n",
      "API 16.6: https://automat.renci.org/chembio/gene/chemical_substance/NCBIGene:7137\n",
      "API 16.6 Automat CHEMBIO API: 0 hits\n",
      "API 7.7: https://automat.renci.org/pharos/gene/chemical_substance/NCBIGene:3558\n",
      "API 7.7 Automat PHAROS API: 5 hits\n",
      "API 17.5: https://automat.renci.org/hmdb/gene/chemical_substance/NCBIGene:3553\n",
      "API 17.5 Automat HMDB API: 1 hits\n",
      "API 4.7: https://automat.renci.org/cord19-scigraph/gene/chemical_substance/NCBIGene:7137\n",
      "API 4.7 Automat CORD19 Scigraph API: 0 hits\n",
      "API 7.8: https://automat.renci.org/pharos/gene/chemical_substance/NCBIGene:3569\n",
      "API 7.8 Automat PHAROS API: 10 hits\n",
      "API 4.8: https://automat.renci.org/cord19-scigraph/gene/chemical_substance/NCBIGene:3569\n",
      "API 4.8 Automat CORD19 Scigraph API: 0 hits\n",
      "API 16.7: https://automat.renci.org/chembio/gene/chemical_substance/NCBIGene:3553\n",
      "API 16.7: https://automat.renci.org/chembio/gene/chemical_substance/NCBIGene:3569\n",
      "API 17.6: https://automat.renci.org/hmdb/gene/chemical_substance/NCBIGene:7124\n",
      "API 17.6 Automat HMDB API: 8 hits\n",
      "API 4.9: https://automat.renci.org/cord19-scigraph/gene/chemical_substance/NCBIGene:7124\n",
      "API 16.7: https://automat.renci.org/chembio/gene/chemical_substance/NCBIGene:7124\n",
      "API 18.5: https://biothings.ncats.io/semmedgene/query?fields=affects (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 17.7: https://automat.renci.org/hmdb/gene/chemical_substance/NCBIGene:3558\n",
      "API 17.7 Automat HMDB API: 0 hits\n",
      "API 17.8: https://automat.renci.org/hmdb/gene/chemical_substance/NCBIGene:3569\n",
      "API 17.8 Automat HMDB API: 0 hits\n",
      "API 17.9: https://automat.renci.org/hmdb/gene/chemical_substance/NCBIGene:7137\n",
      "API 17.9 Automat HMDB API: 0 hits\n",
      "API 2.5: https://automat.renci.org/cord19-scibite/gene/chemical_substance/NCBIGene:7124\n",
      "API 7.9: https://automat.renci.org/pharos/gene/chemical_substance/NCBIGene:7124\n",
      "API 2.5: https://automat.renci.org/cord19-scibite/gene/chemical_substance/NCBIGene:7137\n",
      "API 2.5 Automat CORD19 Scibite API: 1 hits\n",
      "API 2.6: https://automat.renci.org/cord19-scibite/gene/chemical_substance/NCBIGene:3558\n",
      "API 2.6 Automat CORD19 Scibite API: 15 hits\n",
      "API 2.7: https://automat.renci.org/cord19-scibite/gene/chemical_substance/NCBIGene:3569\n",
      "API 2.7 Automat CORD19 Scibite API: 21 hits\n",
      "API 2.8: https://automat.renci.org/cord19-scibite/gene/chemical_substance/NCBIGene:3553\n",
      "API 2.8 Automat CORD19 Scibite API: 18 hits\n",
      "API 18.5 SEMMED Gene API: 0 hits\n",
      "API 16.7 Automat CHEMBIO API: 55 hits\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API 16.8 Automat CHEMBIO API: 0 hits\n",
      "API 4.9 Automat CORD19 Scigraph API: 176 hits\n",
      "API 16.9 Automat CHEMBIO API: 106 hits\n",
      "API 18.6 SEMMED Gene API: 351 hits\n",
      "API 2.9 Automat CORD19 Scibite API: 30 hits\n",
      "API 7.9 Automat PHAROS API: 3 hits\n",
      "API 13.1: https://translator.broadinstitute.org/molepro_reasoner/query (POST -d {'message': {'query_graph': {'edges': [{'id': 'e00', 'source_id': 'n00', 'target_id': 'n01', 'type': 'related_to'}], 'nodes': [{'curie': 'HGNC:6018', 'id': 'n00', 'type': 'gene'}, {'id': 'n01', 'type': 'disease'}]}}})\n",
      "API 13.1 Molecular Data Provider API: 0 hits\n",
      "API 13.2: https://translator.broadinstitute.org/molepro_reasoner/query (POST -d {'message': {'query_graph': {'edges': [{'id': 'e00', 'source_id': 'n00', 'target_id': 'n01', 'type': 'related_to'}], 'nodes': [{'curie': 'HGNC:11947', 'id': 'n00', 'type': 'gene'}, {'id': 'n01', 'type': 'disease'}]}}})\n",
      "API 13.2 Molecular Data Provider API: 0 hits\n",
      "API 14.1: https://translator.broadinstitute.org/genetics_data_provider/query (POST -d {'message': {'query_graph': {'edges': [{'id': 'e00', 'source_id': 'n00', 'target_id': 'n01', 'type': 'associated'}], 'nodes': [{'curie': 'NCBIGene:3569', 'id': 'n00', 'type': 'gene'}, {'id': 'n01', 'type': 'disease'}]}}})\n",
      "API 14.1 Genetics Provider API: 24 hits\n",
      "API 14.2: https://translator.broadinstitute.org/genetics_data_provider/query (POST -d {'message': {'query_graph': {'edges': [{'id': 'e00', 'source_id': 'n00', 'target_id': 'n01', 'type': 'associated'}], 'nodes': [{'curie': 'NCBIGene:7137', 'id': 'n00', 'type': 'gene'}, {'id': 'n01', 'type': 'disease'}]}}})\n",
      "API 14.2 Genetics Provider API: 12 hits\n",
      "API 18.7 SEMMED Gene API: 6 hits\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 297, in callSingleAPI\n",
      "    result = tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/__init__.py\", line 49, in transform\n",
      "    return self.tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/base_transformer.py\", line 78, in transform\n",
      "    item = self.wrap(item)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/cord_transformer.py\", line 16, in wrap\n",
      "    for k in item.keys():\n",
      "RuntimeError: dictionary keys changed during iteration\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API call to CORD Gene API with input ['11892', '5992', '6001', '5417', '11947', '6018'] failed with unknown response\n",
      "API 18.8 SEMMED Gene API: 0 hits\n",
      "API 18.9 SEMMED Gene API: 926 hits\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 297, in callSingleAPI\n",
      "    result = tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/__init__.py\", line 49, in transform\n",
      "    return self.tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/base_transformer.py\", line 78, in transform\n",
      "    item = self.wrap(item)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/cord_transformer.py\", line 16, in wrap\n",
      "    for k in item.keys():\n",
      "RuntimeError: dictionary keys changed during iteration\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API call to CORD Gene API with input ['11892', '5992', '6001', '5417', '11947', '6018'] failed with unknown response\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 297, in callSingleAPI\n",
      "    result = tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/__init__.py\", line 49, in transform\n",
      "    return self.tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/base_transformer.py\", line 78, in transform\n",
      "    item = self.wrap(item)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/cord_transformer.py\", line 16, in wrap\n",
      "    for k in item.keys():\n",
      "RuntimeError: dictionary keys changed during iteration\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API call to CORD Gene API with input ['11892', '5992', '6001', '5417', '11947', '6018'] failed with unknown response\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 297, in callSingleAPI\n",
      "    result = tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/__init__.py\", line 49, in transform\n",
      "    return self.tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/base_transformer.py\", line 78, in transform\n",
      "    item = self.wrap(item)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/cord_transformer.py\", line 16, in wrap\n",
      "    for k in item.keys():\n",
      "RuntimeError: dictionary keys changed during iteration\n",
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 297, in callSingleAPI\n",
      "    result = tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/__init__.py\", line 49, in transform\n",
      "    return self.tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/base_transformer.py\", line 78, in transform\n",
      "    item = self.wrap(item)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/cord_transformer.py\", line 16, in wrap\n",
      "    for k in item.keys():\n",
      "RuntimeError: dictionary keys changed during iteration\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API call to CORD Gene API with input ['11892', '5992', '6001', '5417', '11947', '6018'] failed with unknown response\n",
      "API call to CORD Gene API with input ['11892', '5992', '6001', '5417', '11947', '6018'] failed with unknown response\n",
      "API 1.10: https://biothings.ncats.io/text_mining_targeted_association/query?fields=subject.CHEBI,association.evidence.publications&q=object.NCBIgene:3439 AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 1.10 Text Mining Targeted Association API: 0 hits\n",
      "API 1.11: https://biothings.ncats.io/text_mining_targeted_association/query?fields=subject.CHEBI,association.evidence.publications&q=object.NCBIgene:7137 AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 1.11 Text Mining Targeted Association API: 24 hits\n",
      "API 1.12: https://biothings.ncats.io/text_mining_targeted_association/query?fields=subject.CHEBI,association.evidence.publications&q=object.NCBIgene:3569 AND association.edge_label:negatively_regulates_entity_to_entity&size=1000\n",
      "API 7.10: https://automat.renci.org/pharos/gene/chemical_substance/NCBIGene:7137\n",
      "API 7.10 Automat PHAROS API: 0 hits\n",
      "API 10.10: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3569/diseases?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.10 BioLink API: 12 hits\n",
      "API 17.10: https://automat.renci.org/hmdb/gene/chemical_substance/NCBIGene:3569\n",
      "API 17.10 Automat HMDB API: 0 hits\n",
      "API 17.11: https://automat.renci.org/hmdb/gene/chemical_substance/NCBIGene:7137\n",
      "API 17.11 Automat HMDB API: 1 hits\n",
      "API 7.11: https://automat.renci.org/pharos/gene/chemical_substance/NCBIGene:3569\n",
      "API 7.11 Automat PHAROS API: 1 hits\n",
      "API 16.10: https://automat.renci.org/chembio/gene/chemical_substance/NCBIGene:7137\n",
      "API 16.10 Automat CHEMBIO API: 1 hits\n",
      "API 4.10: https://automat.renci.org/cord19-scigraph/gene/chemical_substance/NCBIGene:7137\n",
      "API 4.10 Automat CORD19 Scigraph API: 0 hits\n",
      "API 17.12: https://automat.renci.org/hmdb/gene/chemical_substance/NCBIGene:3439\n",
      "API 17.12 Automat HMDB API: 0 hits\n",
      "API 2.10: https://automat.renci.org/cord19-scibite/gene/chemical_substance/NCBIGene:7137\n",
      "API 2.10 Automat CORD19 Scibite API: 1 hits\n",
      "API 7.12: https://automat.renci.org/pharos/gene/disease/NCBIGene:7124\n",
      "API 7.12 Automat PHAROS API: 0 hits\n",
      "API 16.11: https://automat.renci.org/chembio/gene/chemical_substance/NCBIGene:3439\n",
      "API 16.11 Automat CHEMBIO API: 4 hits\n",
      "API 7.13: https://automat.renci.org/pharos/gene/disease/NCBIGene:3558\n",
      "API 7.13: https://automat.renci.org/pharos/gene/chemical_substance/NCBIGene:3439\n",
      "API 7.13 Automat PHAROS API: 0 hits\n",
      "API 2.11: https://automat.renci.org/cord19-scibite/gene/chemical_substance/NCBIGene:3439\n",
      "API 2.11 Automat CORD19 Scibite API: 8 hits\n",
      "API 16.12: https://automat.renci.org/chembio/gene/chemical_substance/NCBIGene:3569\n",
      "API 4.11: https://automat.renci.org/cord19-scigraph/gene/chemical_substance/NCBIGene:3569\n",
      "API 4.11 Automat CORD19 Scigraph API: 32 hits\n",
      "API 2.12: https://automat.renci.org/cord19-scibite/gene/disease/NCBIGene:3558\n",
      "API 2.12 Automat CORD19 Scibite API: 0 hits\n",
      "API 2.13: https://automat.renci.org/cord19-scibite/gene/disease/NCBIGene:7124\n",
      "API 2.13: https://automat.renci.org/cord19-scibite/gene/chemical_substance/NCBIGene:3569\n",
      "API 2.13 Automat CORD19 Scibite API: 22 hits\n",
      "API 4.12: https://automat.renci.org/cord19-scigraph/gene/disease/NCBIGene:3558\n",
      "API 4.12 Automat CORD19 Scigraph API: 0 hits\n",
      "API 4.13: https://automat.renci.org/cord19-scigraph/gene/chemical_substance/NCBIGene:3439\n",
      "API 4.13 Automat CORD19 Scigraph API: 0 hits\n",
      "API 4.14: https://automat.renci.org/cord19-scigraph/gene/disease/NCBIGene:7124\n",
      "API 7.14 Automat PHAROS API: 0 hits\n",
      "API 16.12 Automat CHEMBIO API: 66 hits\n",
      "API 2.14 Automat CORD19 Scibite API: 0 hits\n",
      "API 1.12 Text Mining Targeted Association API: 444 hits\n",
      "API 4.14 Automat CORD19 Scigraph API: 0 hits\n",
      "API 6.1: https://biothings.ncats.io/cord_gene/query?fields=associated_with (POST -d q=11892,5992,6001,5417,11947,6018&scopes=hgnc)\n",
      "API 18.10: https://biothings.ncats.io/semmedgene/query?fields=affects (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.10: https://biothings.ncats.io/semmedgene/query?fields=affects (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 10.11: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3558/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.11: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7137/diseases?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.11 BioLink API: 11 hits\n",
      "API 18.10: https://biothings.ncats.io/semmedgene/query?fields=affects (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.10: https://biothings.ncats.io/semmedgene/query?fields=causes (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.10: https://biothings.ncats.io/semmedgene/query?fields=affects (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 10.12 BioLink API: 200 hits\n",
      "API 6.1: https://biothings.ncats.io/cord_gene/query?fields=associated_with (POST -d q=11892,5992,6001,5417,11947,6018&scopes=hgnc)\n",
      "API 10.13: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3439/diseases?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.13 BioLink API: 0 hits\n",
      "API 10.14: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7124/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 18.10 SEMMED Gene API: 3 hits\n",
      "API 10.14 BioLink API: 200 hits\n",
      "API 18.11 SEMMED Gene API: 927 hits\n",
      "API 18.12 SEMMED Gene API: 1132 hits\n",
      "API 18.13 SEMMED Gene API: 50 hits\n",
      "API 18.14 SEMMED Gene API: 2 hits\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 297, in callSingleAPI\n",
      "    result = tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/__init__.py\", line 49, in transform\n",
      "    return self.tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/base_transformer.py\", line 78, in transform\n",
      "    item = self.wrap(item)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/cord_transformer.py\", line 16, in wrap\n",
      "    for k in item.keys():\n",
      "RuntimeError: dictionary keys changed during iteration\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API call to CORD Gene API with input ['11892', '5992', '6001', '5417', '11947', '6018'] failed with unknown response\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\", line 297, in callSingleAPI\n",
      "    result = tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/__init__.py\", line 49, in transform\n",
      "    return self.tf.transform()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/base_transformer.py\", line 78, in transform\n",
      "    item = self.wrap(item)\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/api_response_transform/transformers/cord_transformer.py\", line 16, in wrap\n",
      "    for k in item.keys():\n",
      "RuntimeError: dictionary keys changed during iteration\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API call to CORD Gene API with input ['11892', '5992', '6001', '5417', '11947', '6018'] failed with unknown response\n",
      "API 18.15: https://biothings.ncats.io/semmedgene/query?fields=disrupted_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.15 SEMMED Gene API: 0 hits\n",
      "API 7.15: https://automat.renci.org/pharos/gene/disease/NCBIGene:7124\n",
      "API 7.15 Automat PHAROS API: 3 hits\n",
      "API 7.16: https://automat.renci.org/pharos/gene/disease/NCBIGene:7137\n",
      "API 7.16 Automat PHAROS API: 0 hits\n",
      "API 4.15: https://automat.renci.org/cord19-scigraph/gene/disease/NCBIGene:7137\n",
      "API 4.15 Automat CORD19 Scigraph API: 0 hits\n",
      "API 7.17: https://automat.renci.org/pharos/gene/disease/NCBIGene:3439\n",
      "API 7.17 Automat PHAROS API: 0 hits\n",
      "API 4.16: https://automat.renci.org/cord19-scigraph/gene/disease/NCBIGene:3553\n",
      "API 4.16 Automat CORD19 Scigraph API: 0 hits\n",
      "API 4.17: https://automat.renci.org/cord19-scigraph/gene/disease/NCBIGene:3439\n",
      "API 4.17 Automat CORD19 Scigraph API: 0 hits\n",
      "API 7.18: https://automat.renci.org/pharos/gene/disease/NCBIGene:3569\n",
      "API 4.18: https://automat.renci.org/cord19-scigraph/gene/disease/NCBIGene:3569\n",
      "API 4.18 Automat CORD19 Scigraph API: 0 hits\n",
      "API 2.15: https://automat.renci.org/cord19-scibite/gene/disease/NCBIGene:7124\n",
      "API 2.15: https://automat.renci.org/cord19-scibite/gene/disease/NCBIGene:7137\n",
      "API 2.15 Automat CORD19 Scibite API: 0 hits\n",
      "API 2.16: https://automat.renci.org/cord19-scibite/gene/disease/NCBIGene:3569\n",
      "API 7.18: https://automat.renci.org/pharos/gene/disease/NCBIGene:3553\n",
      "API 2.16: https://automat.renci.org/cord19-scibite/gene/disease/NCBIGene:3439\n",
      "API 2.16 Automat CORD19 Scibite API: 0 hits\n",
      "API 2.17: https://automat.renci.org/cord19-scibite/gene/disease/NCBIGene:3553\n",
      "API 2.17 Automat CORD19 Scibite API: 0 hits\n",
      "API 18.16: https://biothings.ncats.io/semmedgene/query?fields=causes (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 4.19: https://automat.renci.org/cord19-scigraph/gene/disease/NCBIGene:7124\n",
      "API 18.16: https://biothings.ncats.io/semmedgene/query?fields=disrupted_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.16 SEMMED Gene API: 0 hits\n",
      "API 18.17: https://biothings.ncats.io/semmedgene/query?fields=disrupted_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.17 SEMMED Gene API: 0 hits\n",
      "API 18.18: https://biothings.ncats.io/semmedgene/query?fields=disrupted_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.18 SEMMED Gene API: 0 hits\n",
      "API 7.18 Automat PHAROS API: 0 hits\n",
      "API 10.15: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3439/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.15 BioLink API: 35 hits\n",
      "API 2.18 Automat CORD19 Scibite API: 0 hits\n",
      "API 2.19 Automat CORD19 Scibite API: 44 hits\n",
      "API 4.19 Automat CORD19 Scigraph API: 153 hits\n",
      "API 7.19 Automat PHAROS API: 0 hits\n",
      "API 18.19 SEMMED Gene API: 298 hits\n",
      "API 10.16: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3569/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.16: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3553/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.16: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7124/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.16: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7137/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.16 BioLink API: 62 hits\n",
      "API 10.17 BioLink API: 200 hits\n",
      "API 10.18 BioLink API: 0 hits\n",
      "API 10.19 BioLink API: 155 hits\n",
      "API 18.20: https://biothings.ncats.io/semmedgene/query?fields=disrupts (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 2.20: https://automat.renci.org/cord19-scibite/gene/disease/NCBIGene:3558\n",
      "API 2.20: https://automat.renci.org/cord19-scibite/gene/disease/NCBIGene:3553\n",
      "API 2.20 Automat CORD19 Scibite API: 23 hits\n",
      "API 4.20: https://automat.renci.org/cord19-scigraph/gene/disease/NCBIGene:3439\n",
      "API 4.20 Automat CORD19 Scigraph API: 0 hits\n",
      "API 2.21: https://automat.renci.org/cord19-scibite/gene/disease/NCBIGene:3569\n",
      "API 2.21: https://automat.renci.org/cord19-scibite/gene/disease/NCBIGene:7137\n",
      "API 2.21 Automat CORD19 Scibite API: 9 hits\n",
      "API 4.21: https://automat.renci.org/cord19-scigraph/gene/disease/NCBIGene:7137\n",
      "API 4.21 Automat CORD19 Scigraph API: 0 hits\n",
      "API 4.22: https://automat.renci.org/cord19-scigraph/gene/disease/NCBIGene:3553\n",
      "API 4.22 Automat CORD19 Scigraph API: 2 hits\n",
      "API 4.23: https://automat.renci.org/cord19-scigraph/gene/disease/NCBIGene:3558\n",
      "API 4.23 Automat CORD19 Scigraph API: 2 hits\n",
      "API 7.20: https://automat.renci.org/pharos/gene/disease/NCBIGene:3558\n",
      "API 2.22: https://automat.renci.org/cord19-scibite/gene/disease/NCBIGene:3439\n",
      "API 2.22 Automat CORD19 Scibite API: 30 hits\n",
      "API 7.20: https://automat.renci.org/pharos/gene/disease/NCBIGene:3553\n",
      "API 7.20: https://automat.renci.org/pharos/gene/disease/NCBIGene:7137\n",
      "API 7.20 Automat PHAROS API: 4 hits\n",
      "API 7.21: https://automat.renci.org/pharos/gene/disease/NCBIGene:3439\n",
      "API 7.21 Automat PHAROS API: 2 hits\n",
      "API 7.22: https://automat.renci.org/pharos/gene/disease/NCBIGene:3569\n",
      "API 18.20: https://biothings.ncats.io/semmedgene/query?fields=disrupts (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.20: https://biothings.ncats.io/semmedgene/query?fields=disrupts (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 2.23 Automat CORD19 Scibite API: 36 hits\n",
      "API 7.22 Automat PHAROS API: 24 hits\n",
      "API 2.24 Automat CORD19 Scibite API: 24 hits\n",
      "API 10.20: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3439/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.20 BioLink API: 0 hits\n",
      "API 18.20: https://biothings.ncats.io/semmedgene/query?fields=disrupts (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 4.24: https://automat.renci.org/cord19-scigraph/gene/disease/NCBIGene:3569\n",
      "API 7.23 Automat PHAROS API: 99 hits\n",
      "API 10.21: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3569/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 18.20: https://biothings.ncats.io/semmedgene/query?fields=disrupts (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 4.24 Automat CORD19 Scigraph API: 15 hits\n",
      "API 10.21: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3558/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.21: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7137/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.21 BioLink API: 0 hits\n",
      "API 10.22: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3553/interactions?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.22 BioLink API: 0 hits\n",
      "API 18.20 SEMMED Gene API: 4 hits\n",
      "API 7.24 Automat PHAROS API: 42 hits\n",
      "API 10.23 BioLink API: 0 hits\n",
      "API 18.21 SEMMED Gene API: 0 hits\n",
      "API 10.24 BioLink API: 0 hits\n",
      "API 18.22 SEMMED Gene API: 492 hits\n",
      "API 18.23 SEMMED Gene API: 241 hits\n",
      "API 18.24 SEMMED Gene API: 541 hits\n",
      "API 10.25: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3439/phenotypes?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.25 BioLink API: 0 hits\n",
      "API 10.26: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3553/phenotypes?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.26 BioLink API: 1 hits\n",
      "API 10.27: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7124/phenotypes?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.27 BioLink API: 0 hits\n",
      "API 10.28: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3558/phenotypes?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.28 BioLink API: 0 hits\n",
      "API 18.25: https://biothings.ncats.io/semmedgene/query?fields=disrupts (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 10.29: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:7137/phenotypes?direct=True&rows=200&unselect_evidence=True\n",
      "API 18.25: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulated_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.25: https://biothings.ncats.io/semmedgene/query?fields=disrupts (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.25: https://biothings.ncats.io/semmedgene/query?fields=disrupts (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 10.29 BioLink API: 34 hits\n",
      "API 18.25: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulated_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API 18.25 SEMMED Gene API: 1 hits\n",
      "API 18.26 SEMMED Gene API: 18 hits\n",
      "API 18.27 SEMMED Gene API: 2 hits\n",
      "API 18.28 SEMMED Gene API: 0 hits\n",
      "API 18.29 SEMMED Gene API: 0 hits\n",
      "API 10.30: https://api.monarchinitiative.org/api/bioentity/gene/NCBIGene:3569/phenotypes?direct=True&rows=200&unselect_evidence=True\n",
      "API 10.30 BioLink API: 0 hits\n",
      "API 18.30: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulated_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.30: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulated_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.30: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulated_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.30: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulated_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.30: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulated_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.30 SEMMED Gene API: 1 hits\n",
      "API 18.31 SEMMED Gene API: 1 hits\n",
      "API 18.32 SEMMED Gene API: 1734 hits\n",
      "API 18.33 SEMMED Gene API: 2269 hits\n",
      "API 18.34 SEMMED Gene API: 0 hits\n",
      "API 18.35: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulates (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.35: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulated_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.35: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulates (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.35: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulates (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.35: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulates (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.35 SEMMED Gene API: 12 hits\n",
      "API 18.36 SEMMED Gene API: 14 hits\n",
      "API 18.37 SEMMED Gene API: 471 hits\n",
      "API 18.38 SEMMED Gene API: 1 hits\n",
      "API 18.39 SEMMED Gene API: 0 hits\n",
      "API 18.40: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulates (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.40: https://biothings.ncats.io/semmedgene/query?fields=negatively_regulates (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.40: https://biothings.ncats.io/semmedgene/query?fields=physically_interacts_with (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.40: https://biothings.ncats.io/semmedgene/query?fields=physically_interacts_with (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.40: https://biothings.ncats.io/semmedgene/query?fields=physically_interacts_with (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.40 SEMMED Gene API: 0 hits\n",
      "API 18.41 SEMMED Gene API: 1518 hits\n",
      "API 18.42 SEMMED Gene API: 0 hits\n",
      "API 18.43 SEMMED Gene API: 0 hits\n",
      "API 18.44 SEMMED Gene API: 2084 hits\n",
      "API 18.45: https://biothings.ncats.io/semmedgene/query?fields=physically_interacts_with (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.45: https://biothings.ncats.io/semmedgene/query?fields=physically_interacts_with (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.45: https://biothings.ncats.io/semmedgene/query?fields=physically_interacts_with (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.45: https://biothings.ncats.io/semmedgene/query?fields=physically_interacts_with (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.45: https://biothings.ncats.io/semmedgene/query?fields=physically_interacts_with (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.45 SEMMED Gene API: 0 hits\n",
      "API 18.46 SEMMED Gene API: 1 hits\n",
      "API 18.47 SEMMED Gene API: 1 hits\n",
      "API 18.48 SEMMED Gene API: 0 hits\n",
      "API 18.49 SEMMED Gene API: 2977 hits\n",
      "API 18.50: https://biothings.ncats.io/semmedgene/query?fields=positively_regulated_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.50: https://biothings.ncats.io/semmedgene/query?fields=positively_regulated_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.50: https://biothings.ncats.io/semmedgene/query?fields=positively_regulated_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.50: https://biothings.ncats.io/semmedgene/query?fields=positively_regulated_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.50: https://biothings.ncats.io/semmedgene/query?fields=positively_regulated_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.50 SEMMED Gene API: 0 hits\n",
      "API 18.51 SEMMED Gene API: 0 hits\n",
      "API 18.52 SEMMED Gene API: 1864 hits\n",
      "API 18.53 SEMMED Gene API: 0 hits\n",
      "API 18.54 SEMMED Gene API: 0 hits\n",
      "API 18.55: https://biothings.ncats.io/semmedgene/query?fields=positively_regulated_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.55: https://biothings.ncats.io/semmedgene/query?fields=positively_regulates (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.55: https://biothings.ncats.io/semmedgene/query?fields=positively_regulated_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.55: https://biothings.ncats.io/semmedgene/query?fields=positively_regulates (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.55: https://biothings.ncats.io/semmedgene/query?fields=positively_regulated_by (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.55 SEMMED Gene API: 0 hits\n",
      "API 18.56 SEMMED Gene API: 0 hits\n",
      "API 18.57 SEMMED Gene API: 1 hits\n",
      "API 18.58 SEMMED Gene API: 2422 hits\n",
      "API 18.59 SEMMED Gene API: 0 hits\n",
      "API 18.60: https://biothings.ncats.io/semmedgene/query?fields=positively_regulates (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.60: https://biothings.ncats.io/semmedgene/query?fields=positively_regulates (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.60: https://biothings.ncats.io/semmedgene/query?fields=positively_regulates (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.60: https://biothings.ncats.io/semmedgene/query?fields=positively_regulates (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.60: https://biothings.ncats.io/semmedgene/query?fields=positively_regulates (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.60 SEMMED Gene API: 2 hits\n",
      "API 18.61 SEMMED Gene API: 2902 hits\n",
      "API 18.62 SEMMED Gene API: 12 hits\n",
      "API 18.63 SEMMED Gene API: 1023 hits\n",
      "API 18.64 SEMMED Gene API: 0 hits\n",
      "API 18.65: https://biothings.ncats.io/semmedgene/query?fields=related_to (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.65: https://biothings.ncats.io/semmedgene/query?fields=related_to (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.65: https://biothings.ncats.io/semmedgene/query?fields=related_to (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.65: https://biothings.ncats.io/semmedgene/query?fields=prevents (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.65 SEMMED Gene API: 274 hits\n",
      "API 18.66: https://biothings.ncats.io/semmedgene/query?fields=positively_regulates (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.66 SEMMED Gene API: 0 hits\n",
      "API 18.67 SEMMED Gene API: 0 hits\n",
      "API 18.68 SEMMED Gene API: 0 hits\n",
      "API 18.69 SEMMED Gene API: 1 hits\n",
      "API 18.70: https://biothings.ncats.io/semmedgene/query?fields=related_to (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "API 18.70: https://biothings.ncats.io/semmedgene/query?fields=related_to (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.70: https://biothings.ncats.io/semmedgene/query?fields=treats (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.70: https://biothings.ncats.io/semmedgene/query?fields=related_to (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.70: https://biothings.ncats.io/semmedgene/query?fields=related_to (POST -d q=C1415900,C1334112,C1420826,C1334122,C0812246,C0879590&scopes=umls)\n",
      "API 18.70 SEMMED Gene API: 0 hits\n",
      "API 18.71 SEMMED Gene API: 2483 hits\n",
      "API 18.72 SEMMED Gene API: 1309 hits\n",
      "API 18.73 SEMMED Gene API: 0 hits\n",
      "API 18.74 SEMMED Gene API: 4 hits\n",
      "\n",
      "BTE starts to perform id-to-object translation.\n",
      "\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "'Transcript'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-19-36f721aee0da>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0mconfig\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m\"expand\"\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m )\n\u001b[0;32m----> 7\u001b[0;31m \u001b[0mpd3\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconnect\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mverbose\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/query/predict.py\u001b[0m in \u001b[0;36mconnect\u001b[0;34m(self, verbose)\u001b[0m\n\u001b[1;32m    157\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_annotate_edges_with_filters\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mannotatedEdges\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    158\u001b[0m             \u001b[0mdp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mAPIQueryDispatcher\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mannotatedEdges\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mverbose\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mverbose\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 159\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msteps_results\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msyncQuery\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    160\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msteps_results\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msteps_results\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    161\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpt\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprint_query_failure_message\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"results\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mi\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\u001b[0m in \u001b[0;36msyncQuery\u001b[0;34m(self, loop)\u001b[0m\n\u001b[1;32m    409\u001b[0m             \u001b[0mloop\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0masyncio\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_event_loop\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    410\u001b[0m         \u001b[0;31m# return loop.create_task(self.asyncQuery())\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 411\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mloop\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_until_complete\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0masyncQuery\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/nest_asyncio.py\u001b[0m in \u001b[0;36mrun_until_complete\u001b[0;34m(self, future)\u001b[0m\n\u001b[1;32m     93\u001b[0m                 raise RuntimeError(\n\u001b[1;32m     94\u001b[0m                     'Event loop stopped before Future completed.')\n\u001b[0;32m---> 95\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mresult\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     96\u001b[0m         \u001b[0;32mfinally\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     97\u001b[0m             \u001b[0mevents\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_set_running_loop\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mold_running_loop\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/futures.py\u001b[0m in \u001b[0;36mresult\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    176\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__log_traceback\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    177\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_exception\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 178\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_exception\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    179\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_result\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    180\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/tasks.py\u001b[0m in \u001b[0;36m__step\u001b[0;34m(***failed resolving arguments***)\u001b[0m\n\u001b[1;32m    278\u001b[0m                 \u001b[0;31m# We use the `send` method directly, because coroutines\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    279\u001b[0m                 \u001b[0;31m# don't have `__iter__` and `__next__` methods.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 280\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoro\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    281\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    282\u001b[0m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoro\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mthrow\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mexc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\u001b[0m in \u001b[0;36masyncQuery\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    402\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mverbose\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    403\u001b[0m                     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"\\nBTE starts to perform id-to-object translation.\\n\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 404\u001b[0;31m                 \u001b[0mresponses\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mawait\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mannotate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresponses\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    405\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresponses\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    406\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/call_apis/__init__.py\u001b[0m in \u001b[0;36mannotate\u001b[0;34m(self, res, session)\u001b[0m\n\u001b[1;32m    353\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mitem\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mres\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    354\u001b[0m             \u001b[0moutput_ids\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"$association\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"output_type\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0madd\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"$output\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 355\u001b[0;31m         \u001b[0mresolved_ids\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mawait\u001b[0m \u001b[0masyncQuery\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0moutput_ids\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    356\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mitem\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mres\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    357\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mitem\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"$output\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mresolved_ids\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/resolve_ids/__init__.py\u001b[0m in \u001b[0;36masyncQuery\u001b[0;34m(inputIDs, session)\u001b[0m\n\u001b[1;32m     20\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     21\u001b[0m         \u001b[0mdp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mDispatcher\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minputIDs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 22\u001b[0;31m         \u001b[0mdp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdispatch\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     23\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mdp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minvalid\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minvalid\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdict\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     24\u001b[0m             \u001b[0;32mfor\u001b[0m \u001b[0msemantic_type\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcuries\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mdp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minvalid\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/resolve_ids/dispatch.py\u001b[0m in \u001b[0;36mdispatch\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     21\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mdispatch\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     22\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0msemantic_type\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcuries\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minputIds\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 23\u001b[0;31m             \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgenerateQueryTasks\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcuries\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msemantic_type\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     24\u001b[0m             \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mres\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"valid\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mMAX_CONCURRENT_QUERIES_ON_SINGLE_API\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     25\u001b[0m                 self.tasks[i] += res[\"valid\"][\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/resolve_ids/dispatch.py\u001b[0m in \u001b[0;36mgenerateQueryTasks\u001b[0;34m(self, curies, semantic_type)\u001b[0m\n\u001b[1;32m     59\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     60\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mgenerateQueryTasks\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcuries\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msemantic_type\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 61\u001b[0;31m         \u001b[0minputs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgroupIdByPrefix\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcuries\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msemantic_type\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     62\u001b[0m         \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m\"valid\"\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"invalid\"\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     63\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0minputs\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/biothings_explorer/resolve_ids/dispatch.py\u001b[0m in \u001b[0;36mgroupIdByPrefix\u001b[0;34m(self, ids, semantic_type)\u001b[0m\n\u001b[1;32m     44\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     45\u001b[0m                 \u001b[0mprefix\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msplitted\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 46\u001b[0;31m                 \u001b[0;32mif\u001b[0m \u001b[0mprefix\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mID_RESOLVING_APIS\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0msemantic_type\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"mapping\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     47\u001b[0m                     \u001b[0mres\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"invalid\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0madd\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0m_id\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     48\u001b[0m                     \u001b[0;32mcontinue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'Transcript'"
     ]
    }
   ],
   "source": [
    "pd3 = Predict(\n",
    "    input_objs=gene_inputs,\n",
    "    intermediate_nodes=[None], \n",
    "    output_types =['Pathway'], \n",
    "    config={\"expand\": False}\n",
    ")\n",
    "pd3.connect(verbose=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df3 = pd3.display_table_view()\n",
    "df3.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
